<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.551">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Joshua J. Cook, M.S., ACRP-PM, CCRC, Syed Ahzaz H. Shah, B.S., Jacob Hernandez, B.S., Sara Basili, M.S.">
<meta name="dcterms.date" content="2024-04-15">

<title>The Linear Mixed Modelers - Report</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="styles.css">
<meta property="og:title" content="The Linear Mixed Modelers - Report">
<meta property="og:description" content="Inhale, Exhale, Analyze: BMI’s Imprint on Impulse Oscillometry Outcomes">
<meta property="og:site_name" content="The Linear Mixed Modelers">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">The Linear Mixed Modelers</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./literature.html"> 
<span class="menu-text">Literature</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./report.html" aria-current="page"> 
<span class="menu-text">Report</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html"> 
<span class="menu-text">About Us</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><strong>1 Introduction</strong></a></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods"><strong>2 Methods</strong></a>
  <ul class="collapse">
  <li><a href="#mathematical-foundations" id="toc-mathematical-foundations" class="nav-link" data-scroll-target="#mathematical-foundations"><strong>2.1 Mathematical Foundations</strong></a>
  <ul class="collapse">
  <li><a href="#example" id="toc-example" class="nav-link" data-scroll-target="#example">2.1.1 Example</a></li>
  <li><a href="#parameter-estimation" id="toc-parameter-estimation" class="nav-link" data-scroll-target="#parameter-estimation">2.1.1 Parameter Estimation</a></li>
  </ul></li>
  <li><a href="#assumptions" id="toc-assumptions" class="nav-link" data-scroll-target="#assumptions"><strong>2.2 Assumptions</strong></a></li>
  <li><a href="#sample-data-structure" id="toc-sample-data-structure" class="nav-link" data-scroll-target="#sample-data-structure"><strong>2.3 Sample Data Structure</strong></a></li>
  <li><a href="#implementation-in-r" id="toc-implementation-in-r" class="nav-link" data-scroll-target="#implementation-in-r"><strong>2.4 Implementation in R</strong></a>
  <ul class="collapse">
  <li><a href="#analysis-using-lme-function" id="toc-analysis-using-lme-function" class="nav-link" data-scroll-target="#analysis-using-lme-function">2.4.1 Analysis Using lme() Function</a></li>
  <li><a href="#hypothesis-testing" id="toc-hypothesis-testing" class="nav-link" data-scroll-target="#hypothesis-testing">2.4.2 Hypothesis Testing</a></li>
  <li><a href="#model-refinement" id="toc-model-refinement" class="nav-link" data-scroll-target="#model-refinement">2.4.3 Model Refinement</a></li>
  <li><a href="#analysis-using-lmer-function" id="toc-analysis-using-lmer-function" class="nav-link" data-scroll-target="#analysis-using-lmer-function">2.4.4. Analysis Using lmer() Function</a></li>
  <li><a href="#final-model-selection" id="toc-final-model-selection" class="nav-link" data-scroll-target="#final-model-selection">2.4.5 Final Model Selection</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#analysis-and-results" id="toc-analysis-and-results" class="nav-link" data-scroll-target="#analysis-and-results"><strong>3 Analysis and Results</strong></a>
  <ul class="collapse">
  <li><a href="#packages" id="toc-packages" class="nav-link" data-scroll-target="#packages"><strong>3.1 Packages</strong></a></li>
  <li><a href="#data-ingestion" id="toc-data-ingestion" class="nav-link" data-scroll-target="#data-ingestion"><strong>3.2 Data Ingestion</strong></a>
  <ul class="collapse">
  <li><a href="#bmi-from-kaggle-impact-of-bmi-on-ios-measures-on-children-kaggle.com" id="toc-bmi-from-kaggle-impact-of-bmi-on-ios-measures-on-children-kaggle.com" class="nav-link" data-scroll-target="#bmi-from-kaggle-impact-of-bmi-on-ios-measures-on-children-kaggle.com"><strong><code>BMI</code> from Kaggle (</strong>Impact of BMI on IOS measures on children (kaggle.com))</a></li>
  </ul></li>
  <li><a href="#exploratory-data-analysis-eda" id="toc-exploratory-data-analysis-eda" class="nav-link" data-scroll-target="#exploratory-data-analysis-eda"><strong>3.2 Exploratory Data Analysis (EDA)</strong></a>
  <ul class="collapse">
  <li><a href="#missing-values" id="toc-missing-values" class="nav-link" data-scroll-target="#missing-values"><strong>3.2.1 Missing Values</strong></a></li>
  <li><a href="#cleaning-data" id="toc-cleaning-data" class="nav-link" data-scroll-target="#cleaning-data">3.2.2 Cleaning Data</a></li>
  <li><a href="#correlations" id="toc-correlations" class="nav-link" data-scroll-target="#correlations">3.2.3 Correlations</a></li>
  </ul></li>
  <li><a href="#linear-mixed-modeling" id="toc-linear-mixed-modeling" class="nav-link" data-scroll-target="#linear-mixed-modeling"><strong>3.3 Linear Mixed Modeling</strong></a>
  <ul class="collapse">
  <li><a href="#initial-model" id="toc-initial-model" class="nav-link" data-scroll-target="#initial-model"><strong>3.3.1 Initial Model</strong></a></li>
  <li><a href="#imputed-model" id="toc-imputed-model" class="nav-link" data-scroll-target="#imputed-model"><strong>3.3.2 Imputed Model</strong></a></li>
  <li><a href="#final-model" id="toc-final-model" class="nav-link" data-scroll-target="#final-model"><strong>3.3.3 Final Model</strong></a></li>
  <li><a href="#predictions" id="toc-predictions" class="nav-link" data-scroll-target="#predictions"><strong>3.3.4 Predictions</strong></a></li>
  </ul></li>
  </ul></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><strong>4 Conclusion</strong></a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"><strong>5 References</strong></a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Report</h1>
<p class="subtitle lead">Inhale, Exhale, Analyze: BMI’s Imprint on Impulse Oscillometry Outcomes</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Joshua J. Cook, M.S., ACRP-PM, CCRC, Syed Ahzaz H. Shah, B.S., Jacob Hernandez, B.S., Sara Basili, M.S. </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">April 15, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="introduction" class="level1">
<h1><strong>1 Introduction</strong></h1>
<p>Linear mixed-effects models (LMMs) are advanced statistical tools designed to analyze data that exhibit complex structures, such as hierarchical organization, repeated measures, and random effects. These models are particularly useful when data violate the assumptions of traditional ANOVA or regression methods, such as the independence of observations, homoscedasticity, and normality of residuals. LMMs accommodate intra-subject differences, allowing for both fixed effects, which are consistent across individuals, and random effects, which vary among subjects or groups.</p>
<p>The implementation of LMMs has been facilitated by various software packages and programming languages. Brown <span class="citation" data-cites="brown_introduction_2021">(<a href="#ref-brown_introduction_2021" role="doc-biblioref">Brown 2021</a>)</span> provides a comprehensive guide to implementing LMMs in R, a widely used statistical programming language, offering a step-by-step walkthrough of model syntax without delving deeply into complex mathematical foundations. Additionally, the lme4 package, as detailed by Bates et al. <span class="citation" data-cites="bates_fitting_2015">(<a href="#ref-bates_fitting_2015" role="doc-biblioref">Bates et al. 2015</a>)</span>, represents a significant evolution in computational methods for fitting mixed models, offering efficient tools and simplified modeling processes for R users, especially for models with crossed random effects. Pymer4, developed by Jolly <span class="citation" data-cites="jolly_pymer4_2018">(<a href="#ref-jolly_pymer4_2018" role="doc-biblioref">Jolly 2018</a>)</span>, bridges R and Python, offering Python users a flexible and integrated tool for linear mixed modeling by leveraging the capabilities of R’s lme4 package. This tool enhances the analytical capabilities within the Python ecosystem, making advanced statistical methods more accessible to a broader audience.</p>
<p>LMMs find applications across various scientific domains, each with its unique data structures and analytical challenges. The paper by Lee and Shang <span class="citation" data-cites="lee_estimation_nodate">(<a href="#ref-lee_estimation_nodate" role="doc-biblioref">Lee and Shang n.d.</a>)</span> explores the impact of missing data on the estimation and selection in LMMs, highlighting the challenges and proposing a method to record missingness using an indicator-based matrix. This approach is critical for ensuring model accuracy in the presence of missing data, a common issue in real-world datasets. Wang et al. <span class="citation" data-cites="wang_statistical_2022">(<a href="#ref-wang_statistical_2022" role="doc-biblioref">Wang et al. 2022</a>)</span> illustrate the application of LMMs in cardiothoracic surgery outcomes research, using a case study of homograft pulmonary valve replacement data to demonstrate the model’s ability to handle repeated measurements and provide more nuanced understandings of clinical outcomes. Aarts et al <span class="citation" data-cites="aarts_2015">(<a href="#ref-aarts_2015" role="doc-biblioref">Aarts et al. 2015</a>)</span> demonstrates multilevel design experiments in neuroscience and how using linear models on multilevel data can result in increase in false positives. Magezi <span class="citation" data-cites="magezi_linear_2015">(<a href="#ref-magezi_linear_2015" role="doc-biblioref">Magezi 2015</a>)</span> highlights the use of LMMs in within-participant psychology experiments, addressing the complexities of repeated measures and nested data structures common in psychological research. Harrison et al. <span class="citation" data-cites="harrison_brief_2018">(<a href="#ref-harrison_brief_2018" role="doc-biblioref">Harrison et al. 2018</a>)</span> and Bolker et al. <span class="citation" data-cites="bolker_generalized_2009">(<a href="#ref-bolker_generalized_2009" role="doc-biblioref">Bolker et al. 2009</a>)</span> discuss the application of LMMs and generalized linear mixed models (GLMMs) in ecology, emphasizing their utility in analyzing ecological data that involve complex relationships and hierarchical data structures with GRU. Grueber et al <span class="citation" data-cites="grueber_2011">(<a href="#ref-grueber_2011" role="doc-biblioref">GRUEBER et al. 2011</a>)</span> another ecology research, paper focuses on the model averaging and information theorist with LMMS as an alternative to traditional null hypothesis testing. In the medical field, LMMs are employed to model pandemic-induced mortality changes, as demonstrated by Verbeeck et al. <span class="citation" data-cites="verbeeck_linear_2023">(<a href="#ref-verbeeck_linear_2023" role="doc-biblioref">Verbeeck et al. 2023</a>)</span>, and to analyze longitudinal health-related quality of life data in cancer clinical trials, as discussed by Touraine et al. <span class="citation" data-cites="touraine_when_2023">(<a href="#ref-touraine_when_2023" role="doc-biblioref">Touraine et al. 2023</a>)</span>.</p>
<p>The paper “To transform or not to transform: using generalized linear mixed models to analyse reaction time data” by Lo and Andrews <span class="citation" data-cites="lo_transform_2015">(<a href="#ref-lo_transform_2015" role="doc-biblioref">Lo and Andrews 2015</a>)</span> challenges the common practice of transforming reaction time data in cognitive psychology, advocating for GLMMs as a more robust alternative. The “LEVEL” guidelines proposed by Monsalves et al. <span class="citation" data-cites="monsalves_level_2020">(<a href="#ref-monsalves_level_2020" role="doc-biblioref">Monsalves et al. 2020</a>)</span> aim to standardize the reporting of multilevel data and analyses, enhancing comparability across studies. Piepho’s study <span class="citation" data-cites="piepho_analysing_1999">(<a href="#ref-piepho_analysing_1999" role="doc-biblioref">Piepho 1999</a>)</span> on analyzing disease incidence data with GLMMs underscores the inadequacy of traditional methods like ANOVA for such data, highlighting GLMMs’ flexibility. The simulation study by Pusponegoro et al. <span class="citation" data-cites="pusponegoro_linear_2017">(<a href="#ref-pusponegoro_linear_2017" role="doc-biblioref">Pusponegoro et al. 2017</a>)</span> on children’s growth differences emphasizes the importance of choosing the appropriate covariance structure in LMMs for longitudinal data. Lastly, the framework introduced by Steibel et al. <span class="citation" data-cites="steibel_powerful_2009">(<a href="#ref-steibel_powerful_2009" role="doc-biblioref">Steibel et al. 2009</a>)</span> for analyzing RT-PCR data with LMMs showcases the method’s statistical power and flexibility, offering a significant advancement over traditional analysis methods. LMMs are used in a wide array of disciplines, but also in varying study designs, as shown in Table 1.</p>
<p>Table 1. Systematic Review of LMM Use-cases <span class="citation" data-cites="casals_methodological_2014">(<a href="#ref-casals_methodological_2014" role="doc-biblioref">Casals et al. 2014</a>)</span></p>
<p><img src="images/LMM_uses.png" class="img-fluid"></p>
<p>The strengths of LMMs lie in their flexibility to model complex data structures and their ability to handle missing data, making them a powerful tool for a wide range of scientific inquiries. However, their application is not without challenges. Peng and Lu <span class="citation" data-cites="peng_model_2012">(<a href="#ref-peng_model_2012" role="doc-biblioref">Peng and Lu 2012</a>)</span> address the difficulty of variable selection and parameter estimation in LMMs, proposing an iterative procedure to improve model accuracy. Barr <span class="citation" data-cites="barr_random_2013">(<a href="#ref-barr_random_2013" role="doc-biblioref">Barr 2013</a>)</span> critiques existing guidelines for testing interactions within LMMs, proposing new guidelines to ensure more reliable results. The work by Tu <span class="citation" data-cites="tu_using_2015">(<a href="#ref-tu_using_2015" role="doc-biblioref">Tu 2015</a>)</span> on GLMMs for network meta-analyses showcases how mixed models have evolved to tackle complex data, enhancing the accuracy of combining different studies. On the other hand, Fokkema et al. <span class="citation" data-cites="fokkema_generalized_2021">(<a href="#ref-fokkema_generalized_2021" role="doc-biblioref">Marjolein Fokkema and Wolpert 2021</a>)</span> introduce GLMM trees, merging machine learning with mixed models to improve predictions and analysis, particularly useful in mental health research. Despite their robustness, as noted by Schielzeth et al. <span class="citation" data-cites="schielzeth_robustness_2020">(<a href="#ref-schielzeth_robustness_2020" role="doc-biblioref">Schielzeth et al. 2020</a>)</span>, LMMs require careful evaluation of model assumptions and may present computational challenges, especially with high-dimensional datasets.</p>
<p>The literature reviewed here collectively emphasizes the versatility, robustness, and broad applicability of LMMs and GLMMs across various fields of research. Despite their advantages, the importance of careful model selection, acknowledgment of limitations, and the potential need for more complex models such as joint models in certain scenarios are also highlighted. As the use of LMMs continues to grow, the development of standardized processes, such as the LEVEL framework <span class="citation" data-cites="monsalves_level_2020">(<a href="#ref-monsalves_level_2020" role="doc-biblioref">Monsalves et al. 2020</a>)</span> and the 10 protocol put forth by <span class="citation" data-cites="zuur_2016">(<a href="#ref-zuur_2016" role="doc-biblioref">Zuur and Ieno 2016</a>)</span>, and user-friendly tools will be crucial in ensuring the accurate and effective application of these models in research.</p>
</section>
<section id="methods" class="level1">
<h1><strong>2 Methods</strong></h1>
<p>As mentioned by <span class="citation" data-cites="galecki_linear_2014">(<a href="#ref-galecki_linear_2014" role="doc-biblioref">Galecki 2014</a>)</span>, a LMM is:</p>
<blockquote class="blockquote">
<p>a parametric linear model for clustered, longitudinal, or repeated-measures data that quantifies the relationships between a continuous dependent variable and various predictor variables. An LMM may include by <strong>fixed-effect</strong> parameters associated with one or more continuous or categorical covariates and <strong>random effects</strong> associated with one or more random factors.</p>
</blockquote>
<p>Fixed-effect parameters describe the relationships of the covariates to the dependent variable for the entire population. These effects are typically distinct and clearly defined categorical values and are used for classification, such as Gender or Co-morbidities. These effects are commonly utilized in the setting of analysis like ANOVA. Random effects are specific to clusters or subjects within a population. It is typically not possible to include all the distinct levels random effects, but the researcher should always attempt to account for as many random effects as possible to improve the reliability of the LMM.</p>
<p>The selected dataset for this report specifically represent <strong>longitudinal data</strong>, which is data where the dependent variable is measured at several points in time for each unit of analysis. <strong>Participant dropout</strong> is often a concern in the analysis of longitudinal data, with early time points often having a higher compliance rate than later time points. Along with clustered and repeated-measures data, longitudinal data is <strong>hierarchical</strong> because the observations can be placed into hierarchies or levels.</p>
<section id="mathematical-foundations" class="level2">
<h2 class="anchored" data-anchor-id="mathematical-foundations"><strong>2.1 Mathematical Foundations</strong></h2>
<p>LMMs have a mathematical foundation stemming from <strong>linear algebra.</strong> We will be using notation for a 2-level longitudinal model since that is the structure of the dataset in this report. The index <em>i</em> is used to denote participants and <em>t</em> is used to denote the different time points of the observations. Given this notation <em>t</em> is the first level and <em>i</em> is the second level.</p>
<p>Simple LMMs can be defined as in Equation 1.</p>
<p><span id="eq-1"><span class="math display">\[
y=X\beta + Zu+ \epsilon
\tag{1}\]</span></span></p>
<p>where:</p>
<ul>
<li><p>Y is the response vector.</p></li>
<li><p>X is the design matrix for fixed effects.</p></li>
<li><p>β is the vector of fixed effects (parameters associated with the entire population or certain repeatable levels of experimental factors).</p></li>
<li><p>Z is the design matrix for random effects.</p></li>
<li><p><em>u</em> is the vector of random effects (represent random deviations from the population parameters (β ) for different subjects or experimental units; i.e., the variability not explained by the fixed effects).</p></li>
<li><p>ϵ is the vector of residual errors.</p></li>
</ul>
<p>Matrix and Vector Dimensions (Random Intercepts)</p>
<ul>
<li><p>Y is N x 1 matrix where N is the number of the number of repeated measures</p></li>
<li><p>X is a N x p matrix where p is the number of fixed effects covariates</p></li>
<li><p>β is p x 1 column vector</p></li>
<li><p>Z is a N x J matrix where J number of subjects</p></li>
<li><p><em>u</em> is J x 1 vector</p></li>
<li><p>ϵ is n x 1 vector</p></li>
</ul>
<p>For a model with a random intercept, the first column of the X matrix will be all 1s&nbsp;and the first element in the β vector will pertain to that random intercept. The Z matrix in a random intercepts model is a block diagonal matrix, with the block defined by Z<sub>i</sub> matrices.</p>
<p>Adding random effects to the model also changes the size of the dimensions of the Z. If one random effect is added to the matrix then the dimensions change to N x 2q which essentially doubles the columns of the Z matrix to account for the random intercept. <em>u</em> will also double in length to be 2q x 1.</p>
<section id="example" class="level3">
<h3 class="anchored" data-anchor-id="example">2.1.1 Example</h3>
<p>Now let’s go over an example with a 2-level longitudinal structure where we have 100 students with 10 test scores per student and the associated study time for those tests. In this case, the dependent variable is the variable concerning test scores, the fixed effect is the study time and the random effect is the student. For the sake of simplicity, we will only consider a random intercepts model.</p>
<p>Variable Breakdown:</p>
<ul>
<li><p>N=1000: the number of observations which is the number of students multiplied by&nbsp; the number of test scores&nbsp;</p></li>
<li><p>J = 100: the number of students&nbsp;</p></li>
<li><p>p = 2: the random intercept and the fixed effect&nbsp;</p></li>
</ul>
<p>Matrix Notations and Dimension laid out:</p>
<p><span class="math inline">\(Y_{1000\times1} = X_{1000\times 2} \; \beta_{2\times1} + Z_{1000\times100}\;u_{100\times1} + \epsilon_{1000\times1}\)</span></p>
<p>Example Matrices:</p>
<p><span class="math inline">\(y = \begin{bmatrix} Score\\ 75 \\ 80\\ ... \\ 90 \end{bmatrix} X = \begin{bmatrix} Intercept &amp; Study Time \\1 &amp; 2 \\1 &amp; 3\\... &amp; ... \\1 &amp; 5\end{bmatrix}\)</span></p>
<p><span class="math inline">\(\beta = \begin{bmatrix} 1.2\\2.3\end{bmatrix}\)</span></p>
<p>The matrix multiplication can also be broken down into individual equations. In the case of our example we get the following equations:</p>
<p>Level 1 (Time):</p>
<p><span class="math inline">\(Y_{ti} = \beta_{0j} + \beta_{1j} \cdot \text{StudyTime}_{ti} + e_{ti}\)</span></p>
<p>Level 2 (Student):</p>
<p><span class="math inline">\(\beta_{0j} = \gamma_{00} + u_{0j}\)</span></p>
<p>Since this is a random intercepts model, only the intercept equation is needed. γ<sub>00</sub> is the grand intercept mean an u<sub>0j</sub> is the deviation of the j<sub>th</sub> group, which in our case is the student. <span class="citation" data-cites="galecki_linear_2014">(<a href="#ref-galecki_linear_2014" role="doc-biblioref">Galecki 2014</a>)</span>.</p>
</section>
<section id="parameter-estimation" class="level3">
<h3 class="anchored" data-anchor-id="parameter-estimation">2.1.1 Parameter Estimation</h3>
<p>LMMs typically use a Maximum Likelihood estimation or variation called Restricted Maximum likelihood Estimation (REML). Both of these methods obtain parameters of β and θ by optimizing the likelihood function. β are the fixed effects parameters and θ are the covariance matrix parameters where θ depends on the number of random effects and the covariance matrix structure. Our model uses the REML method because it is less biased to the covariance parameters and better at modeling random effects <span class="citation" data-cites="galecki_linear_2014">(<a href="#ref-galecki_linear_2014" role="doc-biblioref">Galecki 2014</a>)</span>.</p>
</section>
</section>
<section id="assumptions" class="level2">
<h2 class="anchored" data-anchor-id="assumptions"><strong>2.2 Assumptions</strong></h2>
<p>Although more flexible than other methods such as ANOVA, there are <strong>several assumptions</strong> for LMMs:</p>
<ol type="1">
<li><p>The relationship between the predictors and response variable is assumed to be linear, within each level of random effects.</p></li>
<li><p>Random effects (<em>u</em>) are assumed to follow a normal distribution with mean zero and variance-covariance matrix G.</p>
<p><span class="math inline">\(\gamma \sim N(0,G)\)</span></p></li>
<li><p>Residual errors (ϵ ) are assumed to follow a normal distribution with mean zero and variance-covariance matrix R.</p>
<p><span class="math inline">\(\epsilon \sim N(0,R)\)</span></p></li>
<li><p>Random effects (<em>u</em>) and residual errors (ϵ ) are assumed to be independent.</p></li>
<li><p>Homoscedasticity is assumed for the residuals across all levels of the independent variables.</p></li>
</ol>
<p>There are several techniques that can be utilized to overcome violations in the LMM assumptions, including variable transformation (to achieve linearity or normality), using robust variance estimates, modifying the structure of random and fixed effects, and employing non-parametric methods or generalized linear mixed models (GLMMs) <span class="citation" data-cites="galecki_linear_2014">(<a href="#ref-galecki_linear_2014" role="doc-biblioref">Galecki 2014</a>)</span>.</p>
</section>
<section id="sample-data-structure" class="level2">
<h2 class="anchored" data-anchor-id="sample-data-structure"><strong>2.3 Sample Data Structure</strong></h2>
<p>LMMs require a tidy data set where each variables are columns and observations are rows. Smaller datasets are usually saved as CSV files and are often loaded from a database. The dataset can contain nulls but they still need to be handled whether they are omitted or imputed and depends on the amount and which columns are affected. The lmer() function in the lme4 library will automatically drop any null values so it is important that data is inspected and visualized before constructing any models. Below is an example of tidy data.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/tidy-4.png" class="img-fluid figure-img"></p>
<figcaption>Figure 1. Tidy data, as defined by Wickham Et. Al.</figcaption>
</figure>
</div>
<p>LMMs also require that the structure of both random and fixed effects be defined before the model is created. The variables that have random variation across groups and those that are fixed must be identified. There are different hierarchies in LMMs. First of all we have to distinguish between clustered and longitudinal data. With cluttered data, as the name suggests, groups the subjects or the unit of analysis into different groups. For a two level data set we can have students be the unit of analysis and the next level up is the classrooms. For a three level data set we can add schools as the third level. Regardless of the amount of levels, the first level is always the subject of the unit of analysis, In the example mentioned above, its students <span class="citation" data-cites="galecki_linear_2014">(<a href="#ref-galecki_linear_2014" role="doc-biblioref">Galecki 2014</a>)</span>.</p>
<p>Then there is also longitudinal data where repeated measures are at the first level and the unit of analysis is the second level. A dataset with different patient cholesterol data over time has the measures at different timepoints as the first level and the patients at a second level <span class="citation" data-cites="galecki_linear_2014">(<a href="#ref-galecki_linear_2014" role="doc-biblioref">Galecki 2014</a>)</span>.</p>
</section>
<section id="implementation-in-r" class="level2">
<h2 class="anchored" data-anchor-id="implementation-in-r"><strong>2.4 Implementation in R</strong></h2>
<p>The implementation begins with importing the dataset into R from a file containing longitudinal retrospective data on the impact of BMI on IOS estimates of airway resistance and reactance in children with sickle cell disease (C-SCD) and African-American children with asthma (C-Asthma). This dataset spans from 2015 to 2020. Data import is executed using the appropriate function, with consideration for specifying file paths and handling header information. Following data importation, preprocessing steps, such as handling missing values and ensuring data integrity, are performed <span class="citation" data-cites="galecki_linear_2014">(<a href="#ref-galecki_linear_2014" role="doc-biblioref">Galecki 2014</a>)</span>.</p>
<section id="analysis-using-lme-function" class="level3">
<h3 class="anchored" data-anchor-id="analysis-using-lme-function">2.4.1 Analysis Using lme() Function</h3>
<p>After preprocessing the data, we proceed with fitting linear mixed-effects models (LMMs) using the lme() function from the <code>nlme</code> package.</p>
<p>The analysis employs the lme() function from the <code>nlme</code> package to fit linear mixed-effects models (LMMs). Model formulation involves specifying a model formula that includes both fixed effects (e.g., BMI, diagnosis of asthma, relevant covariates) and random effects (e.g., random intercepts for subjects). The random argument specifies the random effects structure, while the data argument indicates the dataset to be used. The estimation method (method = “REML”) is specified to use restricted maximum likelihood estimation. It is advantageous to use <code>nlme</code> because it offers a user interface for fitting models with structure in the residuals (including forms of heteroscedasticity and autocorrelation) and in the random-effects covariance matrices).</p>
</section>
<section id="hypothesis-testing" class="level3">
<h3 class="anchored" data-anchor-id="hypothesis-testing">2.4.2 Hypothesis Testing</h3>
<p>Hypotheses are tested to guide model selection and refinement. For instance, Hypothesis 3.1 [1] assesses whether the variance of random effects is greater than zero, while Hypothesis 3.2 [2] investigates the presence of heterogeneous residual variances across treatment groups. These hypotheses are evaluated using likelihood ratio tests or F-tests, depending on the context.</p>
</section>
<section id="model-refinement" class="level3">
<h3 class="anchored" data-anchor-id="model-refinement">2.4.3 Model Refinement</h3>
<p>Based on the outcomes of hypothesis testing and model diagnostics, the model may be refined by removing non-significant fixed effects or selecting an appropriate covariance structure for the residuals. This iterative process entails fitting alternative models and comparing their fit statistics or testing additional hypotheses.</p>
</section>
<section id="analysis-using-lmer-function" class="level3">
<h3 class="anchored" data-anchor-id="analysis-using-lmer-function">2.4.4. Analysis Using lmer() Function</h3>
<p>An alternative approach involves utilizing the lmer() function from the <code>lme4</code> package to fit LMMs. This function follows a similar syntax to lme() but differs in how it handles random effects specification. <code>lme4</code> offers several benefits compared to <code>nlme</code>, including: more efficient linear algebra tools (with associated performance enhancements), simpler syntax and more efficient implementation for fitting models with crossed random effects, implementation of profile likelihood confidence intervals on random-effects parameters, and the ability to fit GLMMs <span class="citation" data-cites="bates_fitting_2015">(<a href="#ref-bates_fitting_2015" role="doc-biblioref">Bates et al. 2015</a>)</span>. Likelihood ratio tests and model diagnostics are employed to assess model fit and inform model selection <span class="citation" data-cites="bates_fitting_2015">(<a href="#ref-bates_fitting_2015" role="doc-biblioref">Bates et al. 2015</a>)</span>.</p>
</section>
<section id="final-model-selection" class="level3">
<h3 class="anchored" data-anchor-id="final-model-selection">2.4.5 Final Model Selection</h3>
<p>The final model is selected based on a synthesis of statistical criteria, including model fit indices, significance of fixed effects, and the adequacy of the model’s assumptions. This selected model is then employed for interpretation and inference concerning the relationships between the predictor variables (e.g., BMI) and the response variable (e.g., IOS measures).</p>
</section>
</section>
</section>
<section id="analysis-and-results" class="level1">
<h1><strong>3 Analysis and Results</strong></h1>
<section id="packages" class="level2">
<h2 class="anchored" data-anchor-id="packages"><strong>3.1 Packages</strong></h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="fu">c</span>(<span class="st">"tidyverse"</span>, <span class="st">"lme4"</span>, <span class="st">"nlme"</span>, <span class="st">"Matrix"</span>, <span class="st">"gt"</span>, <span class="st">"RefManageR"</span>, <span class="st">"DataExplorer"</span>, <span class="st">"gtsummary"</span>, <span class="st">"car"</span>), <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="fu">c</span>(<span class="st">"tidyverse"</span>, <span class="st">"lme4"</span>, <span class="st">"nlme"</span>, <span class="st">"Matrix"</span>, <span class="st">"gt"</span>, <span class="st">"RefManageR"</span>, <span class="st">"DataExplorer"</span>, <span class="st">"gtsummary"</span>, <span class="st">"car"</span>))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lme4)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gt)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gtsummary)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RefManageR)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DataExplorer)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Matrix)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(car)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co">#references &lt;- ReadBib("references.bib")</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co">#summary(references)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ul>
<li><p><code>tidyverse</code>: used for data wrangling and visualization.</p></li>
<li><p><code>lme4</code> and <code>nlme</code>: used for LMM within R.</p></li>
<li><p><code>Matrix</code>: used for matrix manipulation.</p></li>
<li><p><code>gt</code>: used for table generation.</p></li>
<li><p><code>gtsummary</code>: used for summary table generation of descriptive statistics.</p></li>
<li><p><code>RefManageR</code>: used for BibTex reference management.</p></li>
<li><p><code>DataExplorer</code>: used for EDA.</p></li>
<li><p><code>Matrix</code>: used for sparse and dense matrix classes and methods.</p></li>
<li><p><code>car</code>: for qq plots.</p></li>
<li><p><code>reshape2</code>: to reshape date.</p></li>
</ul>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
An error was encountered with the Matrix and lme4 packages during model creation. If this error is encountered, please:
</div>
</div>
<div class="callout-body-container callout-body">
<section id="remove.packagesmatrix" class="level2">
<h2 class="anchored" data-anchor-id="remove.packagesmatrix">remove.packages(“Matrix”)</h2>
</section>
<section id="remove.packageslme4" class="level2">
<h2 class="anchored" data-anchor-id="remove.packageslme4">remove.packages(“lme4”)</h2>
</section>
<section id="install.packageslme4-type-source" class="level2">
<h2 class="anchored" data-anchor-id="install.packageslme4-type-source">install.packages(“lme4”, type = “source”)</h2>
</section>
<section id="librarylme4" class="level2">
<h2 class="anchored" data-anchor-id="librarylme4">library(lme4)</h2>
</section>
</div>
</div>
</section>
<section id="data-ingestion" class="level2">
<h2 class="anchored" data-anchor-id="data-ingestion"><strong>3.2 Data Ingestion</strong></h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the dataset</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>BMI <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/BMI_IOS_SCD_Asthma.csv"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(BMI) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Group"</span>, <span class="st">"Subject_ID"</span>, <span class="st">"Observation_number"</span>, <span class="st">"Hydroxyurea"</span>, <span class="st">"Asthma"</span>, <span class="st">"ICS"</span>, <span class="st">"LABA"</span>, <span class="st">"Gender"</span>, <span class="st">"Age_months"</span>, <span class="st">"Height_cm"</span>, <span class="st">"Weight_Kg"</span>, <span class="st">"BMI"</span>, <span class="st">"R5Hz_PP"</span>, <span class="st">"R20Hz_PP"</span>, <span class="st">"X5Hz_PP"</span>, <span class="st">"Fres_PP"</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>BMI<span class="sc">$</span>Group <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(BMI<span class="sc">$</span>Group)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>BMI<span class="sc">$</span>Subject_ID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(BMI<span class="sc">$</span>Subject_ID)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>BMI<span class="sc">$</span>Observation_number <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(BMI<span class="sc">$</span>Observation_number)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="bmi-from-kaggle-impact-of-bmi-on-ios-measures-on-children-kaggle.com" class="level3">
<h3 class="anchored" data-anchor-id="bmi-from-kaggle-impact-of-bmi-on-ios-measures-on-children-kaggle.com"><strong><code>BMI</code> from Kaggle (</strong><a href="https://www.kaggle.com/datasets/utkarshx27/impact-of-bmi-on-ios-measures">Impact of BMI on IOS measures on children (kaggle.com)</a>)</h3>
<ul>
<li><p><strong>Description</strong>: This dataset is from a retrospective study to assess the impact of BMI on impulse oscillometry (IOS) estimates of airway resistance and reactance in children with sickle cell disease (C-SCD).</p></li>
<li><p><strong>Detailed Description</strong>: The dataset comprises various attributes and measurements across its columns. Categorical variables, such as Group, Subject ID, Observation_number, Hydroxyurea, Asthma, ICS, LABA, and Gender, denote different groupings, individual subjects, and attributes like medication usage and gender. Numerical variables like Age (months), Height (cm), Weight (Kg), BMI, R5Hz_PP, R20Hz_PP, X5Hz_PP, and Fres_PP provide quantitative data on subjects’ characteristics and test results. Notably, the summary also identifies missing values, such as the 14 instances in the Fres_PP variable, which warrant consideration in subsequent analysis. These columns provide measurements and estimates related to airway resistance and reactance obtained using impulse oscillometry (IOS), which is a non-invasive method for assessing respiratory function. These parameters are valuable in understanding the impact of BMI on respiratory measures in children with sickle cell disease (C-SCD) and African-American children with asthma (C-Asthma) participating in the study.</p></li>
<li><p><strong>Why suitable for LMMs</strong>: The dataset has multiple observations, over time, for the same set of participants.</p></li>
</ul>
</section>
</section>
<section id="exploratory-data-analysis-eda" class="level2">
<h2 class="anchored" data-anchor-id="exploratory-data-analysis-eda"><strong>3.2 Exploratory Data Analysis (EDA)</strong></h2>
<p>The structure of the dataframe and variable descriptions are shown in Table 2 and Figure 2. Figures 3-10 systematically explore the features of the data and are described below.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> BMI</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>'data.frame':   219 obs. of  16 variables:
 $ Group             : Factor w/ 2 levels "C-Asthma","C-SCD": 2 2 2 2 2 2 2 2 2 2 ...
 $ Subject_ID        : Factor w/ 90 levels "1","2","3","4",..: 1 1 1 1 2 3 3 4 4 5 ...
 $ Observation_number: Factor w/ 6 levels "1","2","3","4",..: 1 2 3 4 1 1 2 1 2 1 ...
 $ Hydroxyurea       : chr  "Yes" "Yes" "Yes" "Yes" ...
 $ Asthma            : chr  "Yes" "Yes" "Yes" "Yes" ...
 $ ICS               : chr  "Yes" "Yes" "Yes" "Yes" ...
 $ LABA              : chr  "No" "No" "Yes" "Yes" ...
 $ Gender            : chr  "Male" "Male" "Male" "Male" ...
 $ Age_months        : int  239 193 212 224 204 178 186 222 210 196 ...
 $ Height_cm         : num  164 163 164 164 154 ...
 $ Weight_Kg         : num  61.5 62.3 63.1 63.7 66.4 51.9 56.7 66.7 66.9 52.9 ...
 $ BMI               : num  22.8 23.5 23.6 23.7 27.8 ...
 $ R5Hz_PP           : int  145 103 107 87 124 109 117 101 179 136 ...
 $ R20Hz_PP          : int  133 98 98 87 121 86 105 132 153 97 ...
 $ X5Hz_PP           : num  -456 111 174 -303 98 115 107 -216 195 140 ...
 $ Fres_PP           : int  NA 169 159 NA 135 148 159 NA 175 199 ...</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>  Group Subject_ID Observation_number Hydroxyurea Asthma ICS LABA Gender
1 C-SCD          1                  1         Yes    Yes Yes   No   Male
2 C-SCD          1                  2         Yes    Yes Yes   No   Male
3 C-SCD          1                  3         Yes    Yes Yes  Yes   Male
4 C-SCD          1                  4         Yes    Yes Yes  Yes   Male
5 C-SCD          2                  1          No     No  No   No Female
6 C-SCD          3                  1         Yes    Yes  No   No   Male
  Age_months Height_cm Weight_Kg   BMI R5Hz_PP R20Hz_PP X5Hz_PP Fres_PP
1        239     164.1      61.5 22.84     145      133    -456      NA
2        193     162.7      62.3 23.53     103       98     111     169
3        212     163.5      63.1 23.60     107       98     174     159
4        224     163.8      63.7 23.74      87       87    -303      NA
5        204     154.5      66.4 27.82     124      121      98     135
6        178     158.0      51.9 20.79     109       86     115     148</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>variables <span class="ot">&lt;-</span> <span class="fu">colnames</span>(x)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>variables_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">Variable =</span> variables,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">Description =</span> <span class="fu">c</span>(</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column indicates the group to which the subject belongs. There are two groups in the study: children with sickle cell disease (C-SCD) and African-American children with asthma (C-Asthma)."</span>,</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Each subject in the study is assigned a unique identifier or ID, which is listed in this column. The ID is used to differentiate between individual participants."</span>,</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column represents the number assigned to each observation or measurement taken for a particular subject. Since this is a longitudinal study, multiple observations may be recorded for each subject over time."</span>,</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column indicates whether the subject with sickle cell disease (C-SCD) received hydroxyurea treatment. Hydroxyurea is a medication commonly used for the treatment of sickle cell disease."</span>,</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column indicates whether the subject has a diagnosis of asthma. It distinguishes between children with sickle cell disease (C-SCD) and African-American children with asthma (C-Asthma)."</span>,</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column indicates whether the subject is using inhaled corticosteroids (ICS). ICS is a type of medication commonly used for the treatment of asthma and certain other respiratory conditions."</span>,</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column indicates whether the subject is using a long-acting beta-agonist (LABA). LABA is a type of medication often used in combination with inhaled corticosteroids for the treatment of asthma."</span>,</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column represents the gender of the subject, indicating whether they are male or female"</span>,</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column specifies the age of the subject at the time of the observation or measurement. Age is typically measured in months."</span>,</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column represents the height of the subject, typically measured in a standard unit of length, such as centimeters or inches. Height is an important variable to consider in assessing the impact of BMI on respiratory measures."</span>,</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column indicates the weight of the subject at the time of the observation or measurement. Weight is typically measured in kilograms (Kg) and is an important variable for calculating the body mass index (BMI)."</span>,</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Body Mass Index (BMI) is a measure that assesses body weight relative to height. It is calculated by dividing the weight of an individual (in kilograms) by the square of their height (in meters). The BMI column provides the calculated BMI value for each subject based on their weight and height measurements. BMI is commonly used as an indicator of overall body fatness and is often used to classify individuals into different weight categories (e.g., underweight, normal weight, overweight, obese)."</span>,</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column represents the estimate of airway resistance at 5 Hz using impulse oscillometry (IOS). Airway resistance is a measure of the impedance encountered by airflow during respiration. The R5Hz_PP value indicates the airway resistance at the frequency of 5 Hz and is obtained through the IOS testing."</span>,</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column represents the estimate of airway resistance at 20 Hz using impulse oscillometry (IOS). Similar to R5Hz_PP, R20Hz_PP provides the measure of airway resistance at the frequency of 20 Hz based on the IOS testing."</span>,</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column represents the estimate of airway reactance at 5 Hz using impulse oscillometry (IOS). Airway reactance is a measure of the elasticity and stiffness of the airway walls. The X5Hz_PP value indicates the airway reactance at the frequency of 5 Hz and is obtained through the IOS testing."</span>,</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"This column represents the estimate of resonant frequency using impulse oscillometry (IOS). Resonant frequency is a measure of the point at which the reactance of the airways transitions from positive to negative during respiration. The Fres_PP value indicates the resonant frequency and is obtained through the IOS testing.:"</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>variables_table <span class="sc">%&gt;%</span></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>  gt <span class="sc">%&gt;%</span></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tab_header</span>(</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Table 2. Variable Description"</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tab_footnote</span>(</span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">footnote =</span> <span class="st">"Each variable in the dataset, accompanied by a qualitative description from the study team."</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<div id="bwaelapchw" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#bwaelapchw table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#bwaelapchw thead, #bwaelapchw tbody, #bwaelapchw tfoot, #bwaelapchw tr, #bwaelapchw td, #bwaelapchw th {
  border-style: none;
}

#bwaelapchw p {
  margin: 0;
  padding: 0;
}

#bwaelapchw .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#bwaelapchw .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#bwaelapchw .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#bwaelapchw .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#bwaelapchw .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#bwaelapchw .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bwaelapchw .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#bwaelapchw .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#bwaelapchw .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#bwaelapchw .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#bwaelapchw .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#bwaelapchw .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#bwaelapchw .gt_spanner_row {
  border-bottom-style: hidden;
}

#bwaelapchw .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#bwaelapchw .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#bwaelapchw .gt_from_md > :first-child {
  margin-top: 0;
}

#bwaelapchw .gt_from_md > :last-child {
  margin-bottom: 0;
}

#bwaelapchw .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#bwaelapchw .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#bwaelapchw .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#bwaelapchw .gt_row_group_first td {
  border-top-width: 2px;
}

#bwaelapchw .gt_row_group_first th {
  border-top-width: 2px;
}

#bwaelapchw .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#bwaelapchw .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#bwaelapchw .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#bwaelapchw .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bwaelapchw .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#bwaelapchw .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#bwaelapchw .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#bwaelapchw .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#bwaelapchw .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bwaelapchw .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#bwaelapchw .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#bwaelapchw .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#bwaelapchw .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#bwaelapchw .gt_left {
  text-align: left;
}

#bwaelapchw .gt_center {
  text-align: center;
}

#bwaelapchw .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#bwaelapchw .gt_font_normal {
  font-weight: normal;
}

#bwaelapchw .gt_font_bold {
  font-weight: bold;
}

#bwaelapchw .gt_font_italic {
  font-style: italic;
}

#bwaelapchw .gt_super {
  font-size: 65%;
}

#bwaelapchw .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#bwaelapchw .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#bwaelapchw .gt_indent_1 {
  text-indent: 5px;
}

#bwaelapchw .gt_indent_2 {
  text-indent: 10px;
}

#bwaelapchw .gt_indent_3 {
  text-indent: 15px;
}

#bwaelapchw .gt_indent_4 {
  text-indent: 20px;
}

#bwaelapchw .gt_indent_5 {
  text-indent: 25px;
}
</style>

<table class="gt_table table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="header gt_heading">
<th colspan="2" class="gt_heading gt_title gt_font_normal gt_bottom_border">Table 2. Variable Description</th>
</tr>
<tr class="odd gt_col_headings">
<th id="Variable" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Variable</th>
<th id="Description" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Description</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">Group</td>
<td class="gt_row gt_left" headers="Description">This column indicates the group to which the subject belongs. There are two groups in the study: children with sickle cell disease (C-SCD) and African-American children with asthma (C-Asthma).</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">Subject_ID</td>
<td class="gt_row gt_left" headers="Description">Each subject in the study is assigned a unique identifier or ID, which is listed in this column. The ID is used to differentiate between individual participants.</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">Observation_number</td>
<td class="gt_row gt_left" headers="Description">This column represents the number assigned to each observation or measurement taken for a particular subject. Since this is a longitudinal study, multiple observations may be recorded for each subject over time.</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">Hydroxyurea</td>
<td class="gt_row gt_left" headers="Description">This column indicates whether the subject with sickle cell disease (C-SCD) received hydroxyurea treatment. Hydroxyurea is a medication commonly used for the treatment of sickle cell disease.</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">Asthma</td>
<td class="gt_row gt_left" headers="Description">This column indicates whether the subject has a diagnosis of asthma. It distinguishes between children with sickle cell disease (C-SCD) and African-American children with asthma (C-Asthma).</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">ICS</td>
<td class="gt_row gt_left" headers="Description">This column indicates whether the subject is using inhaled corticosteroids (ICS). ICS is a type of medication commonly used for the treatment of asthma and certain other respiratory conditions.</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">LABA</td>
<td class="gt_row gt_left" headers="Description">This column indicates whether the subject is using a long-acting beta-agonist (LABA). LABA is a type of medication often used in combination with inhaled corticosteroids for the treatment of asthma.</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">Gender</td>
<td class="gt_row gt_left" headers="Description">This column represents the gender of the subject, indicating whether they are male or female</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">Age_months</td>
<td class="gt_row gt_left" headers="Description">This column specifies the age of the subject at the time of the observation or measurement. Age is typically measured in months.</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">Height_cm</td>
<td class="gt_row gt_left" headers="Description">This column represents the height of the subject, typically measured in a standard unit of length, such as centimeters or inches. Height is an important variable to consider in assessing the impact of BMI on respiratory measures.</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">Weight_Kg</td>
<td class="gt_row gt_left" headers="Description">This column indicates the weight of the subject at the time of the observation or measurement. Weight is typically measured in kilograms (Kg) and is an important variable for calculating the body mass index (BMI).</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">BMI</td>
<td class="gt_row gt_left" headers="Description">Body Mass Index (BMI) is a measure that assesses body weight relative to height. It is calculated by dividing the weight of an individual (in kilograms) by the square of their height (in meters). The BMI column provides the calculated BMI value for each subject based on their weight and height measurements. BMI is commonly used as an indicator of overall body fatness and is often used to classify individuals into different weight categories (e.g., underweight, normal weight, overweight, obese).</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">R5Hz_PP</td>
<td class="gt_row gt_left" headers="Description">This column represents the estimate of airway resistance at 5 Hz using impulse oscillometry (IOS). Airway resistance is a measure of the impedance encountered by airflow during respiration. The R5Hz_PP value indicates the airway resistance at the frequency of 5 Hz and is obtained through the IOS testing.</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">R20Hz_PP</td>
<td class="gt_row gt_left" headers="Description">This column represents the estimate of airway resistance at 20 Hz using impulse oscillometry (IOS). Similar to R5Hz_PP, R20Hz_PP provides the measure of airway resistance at the frequency of 20 Hz based on the IOS testing.</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">X5Hz_PP</td>
<td class="gt_row gt_left" headers="Description">This column represents the estimate of airway reactance at 5 Hz using impulse oscillometry (IOS). Airway reactance is a measure of the elasticity and stiffness of the airway walls. The X5Hz_PP value indicates the airway reactance at the frequency of 5 Hz and is obtained through the IOS testing.</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">Fres_PP</td>
<td class="gt_row gt_left" headers="Description">This column represents the estimate of resonant frequency using impulse oscillometry (IOS). Resonant frequency is a measure of the point at which the reactance of the airways transitions from positive to negative during respiration. The Fres_PP value indicates the resonant frequency and is obtained through the IOS testing.:</td>
</tr>
</tbody><tfoot class="gt_footnotes">
<tr class="even">
<td colspan="2" class="gt_footnote">Each variable in the dataset, accompanied by a qualitative description from the study team.</td>
</tr>
</tfoot>

</table>

</div>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_str</span>(x)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">introduce</span>(x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>  rows columns discrete_columns continuous_columns all_missing_columns
1  219      16                8                  8                   0
  total_missing_values complete_rows total_observations memory_usage
1                   14           205               3504        34032</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_intro</span>(x, <span class="at">title=</span><span class="st">"Figure 2. Structure of variables and missing observations."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<section id="missing-values" class="level3">
<h3 class="anchored" data-anchor-id="missing-values"><strong>3.2.1 Missing Values</strong></h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_missing</span>(x, <span class="at">title=</span><span class="st">"Figure 3. Breakdown of missing observations."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Based on the missing values count in Figure 3, it appears that there are no missing values in most of the columns, except for Fres_PP, where there are 14 missing values (6.39%). In this case, omitting missing values for Fres_PP is reasonable, considering the small proportion of missing data compared to the total number of observations.</p>
</section>
<section id="cleaning-data" class="level3">
<h3 class="anchored" data-anchor-id="cleaning-data">3.2.2 Cleaning Data</h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 219  16</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>x_clean <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(x) <span class="co"># drops NAs, further analysis is without NA values</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>x_clean<span class="sc">$</span>Gender <span class="ot">&lt;-</span> <span class="fu">tolower</span>(x_clean<span class="sc">$</span>Gender)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(x_clean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 205  16</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(x_clean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>'data.frame':   205 obs. of  16 variables:
 $ Group             : Factor w/ 2 levels "C-Asthma","C-SCD": 2 2 2 2 2 2 2 2 2 2 ...
 $ Subject_ID        : Factor w/ 90 levels "1","2","3","4",..: 1 1 2 3 3 4 5 6 6 7 ...
 $ Observation_number: Factor w/ 6 levels "1","2","3","4",..: 2 3 1 1 2 2 1 1 2 1 ...
 $ Hydroxyurea       : chr  "Yes" "Yes" "No" "Yes" ...
 $ Asthma            : chr  "Yes" "Yes" "No" "Yes" ...
 $ ICS               : chr  "Yes" "Yes" "No" "No" ...
 $ LABA              : chr  "No" "Yes" "No" "No" ...
 $ Gender            : chr  "male" "male" "female" "male" ...
 $ Age_months        : int  193 212 204 178 186 210 196 175 200 173 ...
 $ Height_cm         : num  163 164 154 158 164 ...
 $ Weight_Kg         : num  62.3 63.1 66.4 51.9 56.7 66.9 52.9 39 50.6 56.3 ...
 $ BMI               : num  23.5 23.6 27.8 20.8 20.9 ...
 $ R5Hz_PP           : int  103 107 124 109 117 179 136 96 123 117 ...
 $ R20Hz_PP          : int  98 98 121 86 105 153 97 109 115 117 ...
 $ X5Hz_PP           : num  111 174 98 115 107 195 140 111 164 92 ...
 $ Fres_PP           : int  169 159 135 148 159 175 199 104 109 152 ...
 - attr(*, "na.action")= 'omit' Named int [1:14] 1 4 8 13 18 21 44 46 50 52 ...
  ..- attr(*, "names")= chr [1:14] "1" "4" "8" "13" ...</code></pre>
</div>
</div>
<p><strong>Count Plots for Categorical Variable</strong></p>
<p>The bar plots in Figure 4 show frequency distributions for categories within a distribution. This can aid in data cleaning, checking sparseness or checking for class imbalances. It appears that:</p>
<ul>
<li><p>Most cases are C-SCD compared to C-Asthma (class imbalance).</p></li>
<li><p>The number of observations decreases at subsequent measurements.</p></li>
<li><p>Most cases have Asthma (class imbalance).</p></li>
<li><p>Most cases have LABA (class imbalance).</p></li>
<li><p>Hydroxyurea, ICS, and Gender are relatively evenly distributed.</p></li>
</ul>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_bar</span>(x_clean, <span class="at">title =</span> <span class="st">"Figure 4. Frequency plots of categorical variables."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Histograms</strong></p>
<p>Histograms in Figure 5 show the frequency and distributions of numerical variables This helps identify distribution types among the different variables. Most of the variables below exhibit a normal distribution with some (BMI) showing a slight right skew.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_histogram</span>(x_clean, <span class="at">title =</span> <span class="st">"Figure 5. Histogram plots of numerical variables."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Q-Q Plots</strong></p>
<p>The QQ plots in Figure 6 serve as a visual aid to asses normality in the covariates. The closer the points are to the straight diagonal line, the more “normal” the data is distributed. Most of the variables show a normal distribution. BMI has a substantial amount of points on the right corner of the plot that go off of the diagonal line potentially representing a non-normal distribution. Weight_Kg shows a similar skew.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_qq</span>(<span class="fu">na.omit</span>(x), <span class="at">title =</span> <span class="st">"Figure 6. QQ plots to assess normality of numerical variables."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Principal Component Analysis (PCA)</strong></p>
<p>The PCA plots in Figure 7 show the numerical variables in our data set split into principal components. More than half (54.8%) of the variance can be explained with just 4 principal components. This can be useful if we want to simplify our model by only keeping the principal components that explain most of the variance.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_prcomp</span>(<span class="fu">na.omit</span>(x), <span class="at">title =</span> <span class="st">"Figure 7. PCA to assess key principle components that explain the variance."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-9-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Box Plots</strong></p>
<p>Based on the boxplots in Figure 8, it’s evident that all variables except “Age (months)” and “Height (cm)” contain outliers. Now, let’s pinpoint these outliers and calculate summary statistics (Table 3).</p>
<p>Figure 8. Boxplots of numerical variables.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>numeric_vars <span class="ot">&lt;-</span> x_clean <span class="sc">%&gt;%</span> </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select_if</span>(is.numeric)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Boxplot for each numeric variable</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>))</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (col <span class="cf">in</span> <span class="fu">colnames</span>(numeric_vars)) {</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">boxplot</span>(numeric_vars[[col]], <span class="at">main=</span>col)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-10-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding a general title for the entire set of boxplots</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="co">#mtext("Figure 8. Box plots of numerical variables.", side=3, line=1, outer=TRUE, cex=1.5)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a function to detect outliers in each column</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>detect_outliers <span class="ot">&lt;-</span> <span class="cf">function</span>(column) {</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  Q1 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(column, <span class="fl">0.25</span>)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  Q3 <span class="ot">&lt;-</span> <span class="fu">quantile</span>(column, <span class="fl">0.75</span>)</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  IQR <span class="ot">&lt;-</span> Q3 <span class="sc">-</span> Q1</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  lower_bound <span class="ot">&lt;-</span> Q1 <span class="sc">-</span> <span class="fl">1.5</span> <span class="sc">*</span> IQR</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  upper_bound <span class="ot">&lt;-</span> Q3 <span class="sc">+</span> <span class="fl">1.5</span> <span class="sc">*</span> IQR</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  outliers <span class="ot">&lt;-</span> column[column <span class="sc">&lt;</span> lower_bound <span class="sc">|</span> column <span class="sc">&gt;</span> upper_bound]</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(outliers)</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Iterate over each column and print outliers; not removed</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (col <span class="cf">in</span> <span class="fu">names</span>(numeric_vars)) {</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>  outliers <span class="ot">&lt;-</span> <span class="fu">detect_outliers</span>(numeric_vars[[col]])</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(outliers) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"Outliers in"</span>, col, <span class="st">":</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(outliers)</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Outliers in Weight_Kg :
 [1]  91.4 100.2 104.4 111.2 120.4 108.6 105.1 100.0  95.5 130.1 108.5 124.6
[13]  95.2

Outliers in BMI :
 [1] 27.82 37.85 41.49 42.09 26.65 28.66 41.85 41.18 39.89 40.05 38.58 38.74
[13] 29.92 47.21 41.34 48.07 27.82 26.95

Outliers in R5Hz_PP :
[1] 187 183 196  10 199 198

Outliers in R20Hz_PP :
[1] 153 150  12 162 153

Outliers in X5Hz_PP :
 [1]  439 1117  260  270  297  465  338  285  306  -68  543  381</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>x_clean <span class="sc">%&gt;%</span> </span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tbl_summary</span>( <span class="co">#gtSummary Table</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">by=</span>Group,</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="fu">list</span>(</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">c</span>(<span class="st">'Age_months'</span>, <span class="st">'Height_cm'</span>, <span class="st">'Weight_Kg'</span>, <span class="st">'BMI'</span>, <span class="st">'R5Hz_PP'</span>, <span class="st">'R20Hz_PP'</span>, <span class="st">'X5Hz_PP'</span>, <span class="st">'Fres_PP'</span>) <span class="sc">~</span> <span class="st">'continuous2'</span>),</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">statistic =</span> <span class="fu">all_continuous2</span>() <span class="sc">~</span> <span class="fu">c</span>(</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"{mean} ± {sd}"</span>,</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"{median} ({p25}, {p75})"</span>,</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>                       <span class="st">"{min}, {max}"</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>                       ),</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">digits =</span> <span class="fu">all_continuous2</span>() <span class="sc">~</span> <span class="dv">2</span>,</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">missing=</span><span class="st">"ifany"</span>,</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>  bold_labels <span class="sc">%&gt;%</span></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">italicize_levels</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_gt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tab_header</span>(</span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Table. 3 Summary Statistics"</span></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tab_footnote</span>(</span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">footnote =</span> <span class="st">"Summary statistics for all variables."</span></span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<div id="eyisuxepwz" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#eyisuxepwz table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#eyisuxepwz thead, #eyisuxepwz tbody, #eyisuxepwz tfoot, #eyisuxepwz tr, #eyisuxepwz td, #eyisuxepwz th {
  border-style: none;
}

#eyisuxepwz p {
  margin: 0;
  padding: 0;
}

#eyisuxepwz .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#eyisuxepwz .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#eyisuxepwz .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#eyisuxepwz .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#eyisuxepwz .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#eyisuxepwz .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#eyisuxepwz .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#eyisuxepwz .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#eyisuxepwz .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#eyisuxepwz .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#eyisuxepwz .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#eyisuxepwz .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#eyisuxepwz .gt_spanner_row {
  border-bottom-style: hidden;
}

#eyisuxepwz .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#eyisuxepwz .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#eyisuxepwz .gt_from_md > :first-child {
  margin-top: 0;
}

#eyisuxepwz .gt_from_md > :last-child {
  margin-bottom: 0;
}

#eyisuxepwz .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#eyisuxepwz .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#eyisuxepwz .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#eyisuxepwz .gt_row_group_first td {
  border-top-width: 2px;
}

#eyisuxepwz .gt_row_group_first th {
  border-top-width: 2px;
}

#eyisuxepwz .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#eyisuxepwz .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#eyisuxepwz .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#eyisuxepwz .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#eyisuxepwz .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#eyisuxepwz .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#eyisuxepwz .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#eyisuxepwz .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#eyisuxepwz .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#eyisuxepwz .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#eyisuxepwz .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#eyisuxepwz .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#eyisuxepwz .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#eyisuxepwz .gt_left {
  text-align: left;
}

#eyisuxepwz .gt_center {
  text-align: center;
}

#eyisuxepwz .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#eyisuxepwz .gt_font_normal {
  font-weight: normal;
}

#eyisuxepwz .gt_font_bold {
  font-weight: bold;
}

#eyisuxepwz .gt_font_italic {
  font-style: italic;
}

#eyisuxepwz .gt_super {
  font-size: 65%;
}

#eyisuxepwz .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#eyisuxepwz .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#eyisuxepwz .gt_indent_1 {
  text-indent: 5px;
}

#eyisuxepwz .gt_indent_2 {
  text-indent: 10px;
}

#eyisuxepwz .gt_indent_3 {
  text-indent: 15px;
}

#eyisuxepwz .gt_indent_4 {
  text-indent: 20px;
}

#eyisuxepwz .gt_indent_5 {
  text-indent: 25px;
}
</style>

<table class="gt_table table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="header gt_heading">
<th colspan="3" class="gt_heading gt_title gt_font_normal gt_bottom_border">Table. 3 Summary Statistics</th>
</tr>
<tr class="odd gt_col_headings">
<th id="<strong>Characteristic</strong>" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col"><strong>Characteristic</strong></th>
<th id="<strong>C-Asthma</strong>, N = 56<span class=&quot;gt_footnote_marks&quot; style=&quot;white-space:nowrap;font-style:italic;font-weight:normal;&quot;><sup>1</sup></span>" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col"><strong>C-Asthma</strong>, N = 56<span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"><sup>1</sup></span></th>
<th id="<strong>C-SCD</strong>, N = 149<span class=&quot;gt_footnote_marks&quot; style=&quot;white-space:nowrap;font-style:italic;font-weight:normal;&quot;><sup>1</sup></span>" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col"><strong>C-SCD</strong>, N = 149<span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"><sup>1</sup></span></th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">Observation_number</td>
<td class="gt_row gt_center" headers="stat_1"></td>
<td class="gt_row gt_center" headers="stat_2"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;1</td>
<td class="gt_row gt_center" headers="stat_1">34 (61%)</td>
<td class="gt_row gt_center" headers="stat_2">51 (34%)</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;2</td>
<td class="gt_row gt_center" headers="stat_1">13 (23%)</td>
<td class="gt_row gt_center" headers="stat_2">44 (30%)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;3</td>
<td class="gt_row gt_center" headers="stat_1">7 (13%)</td>
<td class="gt_row gt_center" headers="stat_2">31 (21%)</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;4</td>
<td class="gt_row gt_center" headers="stat_1">2 (3.6%)</td>
<td class="gt_row gt_center" headers="stat_2">17 (11%)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;5</td>
<td class="gt_row gt_center" headers="stat_1">0 (0%)</td>
<td class="gt_row gt_center" headers="stat_2">5 (3.4%)</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;6</td>
<td class="gt_row gt_center" headers="stat_1">0 (0%)</td>
<td class="gt_row gt_center" headers="stat_2">1 (0.7%)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">Hydroxyurea</td>
<td class="gt_row gt_center" headers="stat_1">0 (0%)</td>
<td class="gt_row gt_center" headers="stat_2">112 (75%)</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">Asthma</td>
<td class="gt_row gt_center" headers="stat_1">56 (100%)</td>
<td class="gt_row gt_center" headers="stat_2">106 (71%)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">ICS</td>
<td class="gt_row gt_center" headers="stat_1">40 (71%)</td>
<td class="gt_row gt_center" headers="stat_2">73 (49%)</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">LABA</td>
<td class="gt_row gt_center" headers="stat_1">24 (43%)</td>
<td class="gt_row gt_center" headers="stat_2">14 (9.4%)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">Gender</td>
<td class="gt_row gt_center" headers="stat_1"></td>
<td class="gt_row gt_center" headers="stat_2"></td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;female</td>
<td class="gt_row gt_center" headers="stat_1">22 (39%)</td>
<td class="gt_row gt_center" headers="stat_2">59 (40%)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;male</td>
<td class="gt_row gt_center" headers="stat_1">34 (61%)</td>
<td class="gt_row gt_center" headers="stat_2">90 (60%)</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">Age_months</td>
<td class="gt_row gt_center" headers="stat_1"></td>
<td class="gt_row gt_center" headers="stat_2"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Mean ± SD</td>
<td class="gt_row gt_center" headers="stat_1">132.98 ± 35.59</td>
<td class="gt_row gt_center" headers="stat_2">142.16 ± 44.70</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Median (IQR)</td>
<td class="gt_row gt_center" headers="stat_1">134.00 (100.00, 147.25)</td>
<td class="gt_row gt_center" headers="stat_2">144.00 (105.00, 179.00)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Range</td>
<td class="gt_row gt_center" headers="stat_1">87.00, 213.00</td>
<td class="gt_row gt_center" headers="stat_2">50.00, 215.00</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">Height_cm</td>
<td class="gt_row gt_center" headers="stat_1"></td>
<td class="gt_row gt_center" headers="stat_2"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Mean ± SD</td>
<td class="gt_row gt_center" headers="stat_1">146.57 ± 17.33</td>
<td class="gt_row gt_center" headers="stat_2">145.00 ± 18.83</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Median (IQR)</td>
<td class="gt_row gt_center" headers="stat_1">144.50 (131.00, 162.00)</td>
<td class="gt_row gt_center" headers="stat_2">146.20 (132.00, 160.30)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Range</td>
<td class="gt_row gt_center" headers="stat_1">120.00, 185.00</td>
<td class="gt_row gt_center" headers="stat_2">101.80, 178.90</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">Weight_Kg</td>
<td class="gt_row gt_center" headers="stat_1"></td>
<td class="gt_row gt_center" headers="stat_2"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Mean ± SD</td>
<td class="gt_row gt_center" headers="stat_1">52.98 ± 30.88</td>
<td class="gt_row gt_center" headers="stat_2">40.11 ± 16.06</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Median (IQR)</td>
<td class="gt_row gt_center" headers="stat_1">38.30 (30.60, 66.83)</td>
<td class="gt_row gt_center" headers="stat_2">36.70 (28.30, 51.90)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Range</td>
<td class="gt_row gt_center" headers="stat_1">19.60, 130.10</td>
<td class="gt_row gt_center" headers="stat_2">14.90, 104.40</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">BMI</td>
<td class="gt_row gt_center" headers="stat_1"></td>
<td class="gt_row gt_center" headers="stat_2"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Mean ± SD</td>
<td class="gt_row gt_center" headers="stat_1">23.02 ± 9.17</td>
<td class="gt_row gt_center" headers="stat_2">18.32 ± 4.10</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Median (IQR)</td>
<td class="gt_row gt_center" headers="stat_1">18.38 (17.17, 25.66)</td>
<td class="gt_row gt_center" headers="stat_2">17.59 (15.87, 19.48)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Range</td>
<td class="gt_row gt_center" headers="stat_1">14.10, 48.07</td>
<td class="gt_row gt_center" headers="stat_2">13.70, 42.09</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">R5Hz_PP</td>
<td class="gt_row gt_center" headers="stat_1"></td>
<td class="gt_row gt_center" headers="stat_2"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Mean ± SD</td>
<td class="gt_row gt_center" headers="stat_1">89.57 ± 25.09</td>
<td class="gt_row gt_center" headers="stat_2">107.67 ± 32.83</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Median (IQR)</td>
<td class="gt_row gt_center" headers="stat_1">85.50 (72.75, 104.50)</td>
<td class="gt_row gt_center" headers="stat_2">105.00 (85.00, 125.00)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Range</td>
<td class="gt_row gt_center" headers="stat_1">43.00, 168.00</td>
<td class="gt_row gt_center" headers="stat_2">10.00, 199.00</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">R20Hz_PP</td>
<td class="gt_row gt_center" headers="stat_1"></td>
<td class="gt_row gt_center" headers="stat_2"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Mean ± SD</td>
<td class="gt_row gt_center" headers="stat_1">77.16 ± 21.70</td>
<td class="gt_row gt_center" headers="stat_2">88.77 ± 24.61</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Median (IQR)</td>
<td class="gt_row gt_center" headers="stat_1">72.00 (62.00, 88.25)</td>
<td class="gt_row gt_center" headers="stat_2">88.00 (72.00, 102.00)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Range</td>
<td class="gt_row gt_center" headers="stat_1">38.00, 135.00</td>
<td class="gt_row gt_center" headers="stat_2">12.00, 162.00</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">X5Hz_PP</td>
<td class="gt_row gt_center" headers="stat_1"></td>
<td class="gt_row gt_center" headers="stat_2"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Mean ± SD</td>
<td class="gt_row gt_center" headers="stat_1">101.87 ± 51.52</td>
<td class="gt_row gt_center" headers="stat_2">138.04 ± 115.20</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Median (IQR)</td>
<td class="gt_row gt_center" headers="stat_1">96.00 (74.75, 117.75)</td>
<td class="gt_row gt_center" headers="stat_2">116.00 (83.00, 157.00)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Range</td>
<td class="gt_row gt_center" headers="stat_1">1.69, 381.00</td>
<td class="gt_row gt_center" headers="stat_2">-68.00, 1,117.00</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-weight: bold">Fres_PP</td>
<td class="gt_row gt_center" headers="stat_1"></td>
<td class="gt_row gt_center" headers="stat_2"></td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Mean ± SD</td>
<td class="gt_row gt_center" headers="stat_1">120.55 ± 31.25</td>
<td class="gt_row gt_center" headers="stat_2">136.75 ± 33.35</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Median (IQR)</td>
<td class="gt_row gt_center" headers="stat_1">111.50 (99.00, 142.50)</td>
<td class="gt_row gt_center" headers="stat_2">132.00 (110.00, 163.00)</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="label" style="font-style: italic">&nbsp;&nbsp;&nbsp;&nbsp;Range</td>
<td class="gt_row gt_center" headers="stat_1">61.00, 236.00</td>
<td class="gt_row gt_center" headers="stat_2">66.00, 225.00</td>
</tr>
</tbody><tfoot class="gt_footnotes">
<tr class="even">
<td colspan="3" class="gt_footnote">Summary statistics for all variables.</td>
</tr>
<tr class="odd">
<td colspan="3" class="gt_footnote"><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"><sup>1</sup></span> n (%)</td>
</tr>
</tfoot>

</table>

</div>
</div>
</div>
</div>
<p><strong>Participant Dropout</strong></p>
<p>Figure 9 and Table 4 show how many subjects had data at each subsequent timepoint, which suggests that this study experienced significant participant dropout over time. This dropout may or may not be attributed to the study itself and should be investigated further. A strength of LMM is that it can handle unbalanced groups (i.e., patients), so we will continue with modeling regardless.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>x_clean_timepoints <span class="ot">&lt;-</span> x_clean <span class="sc">%&gt;%</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Observation_number) <span class="sc">%&gt;%</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Unique_Subjects =</span> <span class="fu">n_distinct</span>(Subject_ID))</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>x_clean_timepoints<span class="sc">$</span>Unique_Subjects <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(x_clean_timepoints<span class="sc">$</span>Unique_Subjects)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(x_clean_timepoints, <span class="fu">aes</span>(<span class="at">x =</span> Observation_number, <span class="at">y =</span> Unique_Subjects)) <span class="sc">+</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span> <span class="co"># Add points for each observation</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="dv">1</span>), <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span> <span class="co"># Connect the points with a line</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Figure 9. Participant dropout over time."</span>,</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Timepoint"</span>,</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Number of Unique Subjects"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>x_clean_timepoints <span class="sc">%&gt;%</span> </span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gt</span>() <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tab_header</span>(</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Table 4. Number of participants at each timepoint."</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tab_footnote</span>(</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">footnote =</span> <span class="st">"Counts of unique subjects reveal an increasing amount of missing data at subsequent observation visits."</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<div id="zctemosbeg" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#zctemosbeg table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#zctemosbeg thead, #zctemosbeg tbody, #zctemosbeg tfoot, #zctemosbeg tr, #zctemosbeg td, #zctemosbeg th {
  border-style: none;
}

#zctemosbeg p {
  margin: 0;
  padding: 0;
}

#zctemosbeg .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#zctemosbeg .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#zctemosbeg .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#zctemosbeg .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#zctemosbeg .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#zctemosbeg .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#zctemosbeg .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#zctemosbeg .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#zctemosbeg .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#zctemosbeg .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#zctemosbeg .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#zctemosbeg .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#zctemosbeg .gt_spanner_row {
  border-bottom-style: hidden;
}

#zctemosbeg .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#zctemosbeg .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#zctemosbeg .gt_from_md > :first-child {
  margin-top: 0;
}

#zctemosbeg .gt_from_md > :last-child {
  margin-bottom: 0;
}

#zctemosbeg .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#zctemosbeg .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#zctemosbeg .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#zctemosbeg .gt_row_group_first td {
  border-top-width: 2px;
}

#zctemosbeg .gt_row_group_first th {
  border-top-width: 2px;
}

#zctemosbeg .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#zctemosbeg .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#zctemosbeg .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#zctemosbeg .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#zctemosbeg .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#zctemosbeg .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#zctemosbeg .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#zctemosbeg .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#zctemosbeg .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#zctemosbeg .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#zctemosbeg .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#zctemosbeg .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#zctemosbeg .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#zctemosbeg .gt_left {
  text-align: left;
}

#zctemosbeg .gt_center {
  text-align: center;
}

#zctemosbeg .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#zctemosbeg .gt_font_normal {
  font-weight: normal;
}

#zctemosbeg .gt_font_bold {
  font-weight: bold;
}

#zctemosbeg .gt_font_italic {
  font-style: italic;
}

#zctemosbeg .gt_super {
  font-size: 65%;
}

#zctemosbeg .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#zctemosbeg .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#zctemosbeg .gt_indent_1 {
  text-indent: 5px;
}

#zctemosbeg .gt_indent_2 {
  text-indent: 10px;
}

#zctemosbeg .gt_indent_3 {
  text-indent: 15px;
}

#zctemosbeg .gt_indent_4 {
  text-indent: 20px;
}

#zctemosbeg .gt_indent_5 {
  text-indent: 25px;
}
</style>

<table class="gt_table table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="header gt_heading">
<th colspan="2" class="gt_heading gt_title gt_font_normal gt_bottom_border">Table 4. Number of participants at each timepoint.</th>
</tr>
<tr class="odd gt_col_headings">
<th id="Observation_number" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">Observation_number</th>
<th id="Unique_Subjects" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Unique_Subjects</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_center" headers="Observation_number">1</td>
<td class="gt_row gt_right" headers="Unique_Subjects">85</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="Observation_number">2</td>
<td class="gt_row gt_right" headers="Unique_Subjects">57</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="Observation_number">3</td>
<td class="gt_row gt_right" headers="Unique_Subjects">38</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="Observation_number">4</td>
<td class="gt_row gt_right" headers="Unique_Subjects">19</td>
</tr>
<tr class="odd">
<td class="gt_row gt_center" headers="Observation_number">5</td>
<td class="gt_row gt_right" headers="Unique_Subjects">5</td>
</tr>
<tr class="even">
<td class="gt_row gt_center" headers="Observation_number">6</td>
<td class="gt_row gt_right" headers="Unique_Subjects">1</td>
</tr>
</tbody><tfoot class="gt_footnotes">
<tr class="even">
<td colspan="2" class="gt_footnote">Counts of unique subjects reveal an increasing amount of missing data at subsequent observation visits.</td>
</tr>
</tfoot>

</table>

</div>
</div>
</div>
</div>
</section>
<section id="correlations" class="level3">
<h3 class="anchored" data-anchor-id="correlations">3.2.3 Correlations</h3>
<p>Figure 10 highlights correlations between variables that should be assessed before any modeling. Age (months) and Height (cm): There is a strong positive correlation (0.914) between Age (months) and Height (cm). This implies that as age increases, height tends to increase as well. This correlation is expected, as children tend to grow taller as they get older.</p>
<p>Weight (Kg) and BMI: There is a strong positive correlation (0.927) between Weight (Kg) and BMI. This suggests that as weight increases, BMI (Body Mass Index) tends to increase as well. This correlation is expected because BMI is calculated using weight and height measurements.</p>
<p>R5Hz_PP and Fres_PP: There is a strong positive correlation (0.754) between R5Hz_PP and Fres_PP.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_correlation</span>(<span class="fu">na.omit</span>(x), <span class="at">maxcat=</span><span class="dv">5</span>L, <span class="at">title =</span> <span class="st">"Figure 10. Correlation matrix of all variables."</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>correlation_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(numeric_vars)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(correlation_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>           Age_months Height_cm Weight_Kg           BMI      R5Hz_PP   R20Hz_PP
Age_months  1.0000000 0.9136787 0.6972936  4.441633e-01 4.038596e-01 0.33200111
Height_cm   0.9136787 1.0000000 0.7478733  4.581481e-01 3.100369e-01 0.24291582
Weight_Kg   0.6972936 0.7478733 1.0000000  9.265856e-01 1.261621e-01 0.10391395
BMI         0.4441633 0.4581481 0.9265856  1.000000e+00 6.430664e-06 0.01337894
R5Hz_PP     0.4038596 0.3100369 0.1261621  6.430664e-06 1.000000e+00 0.70961067
R20Hz_PP    0.3320011 0.2429158 0.1039139  1.337894e-02 7.096107e-01 1.00000000
X5Hz_PP     0.3885715 0.3745554 0.1346387 -2.990486e-02 4.464806e-01 0.20055148
Fres_PP     0.5876287 0.5214880 0.1975967 -1.523383e-02 7.538067e-01 0.54826299
               X5Hz_PP     Fres_PP
Age_months  0.38857148  0.58762865
Height_cm   0.37455537  0.52148796
Weight_Kg   0.13463870  0.19759669
BMI        -0.02990486 -0.01523383
R5Hz_PP     0.44648059  0.75380665
R20Hz_PP    0.20055148  0.54826299
X5Hz_PP     1.00000000  0.50290470
Fres_PP     0.50290470  1.00000000</code></pre>
</div>
</div>
</section>
</section>
<section id="linear-mixed-modeling" class="level2">
<h2 class="anchored" data-anchor-id="linear-mixed-modeling"><strong>3.3 Linear Mixed Modeling</strong></h2>
<p>In this dataset, the variable of interest is body mass index (BMI). Additionally, controlled variables are present such as group, age, weight, height, and other co-morbidities. These are the fixed effects. On the other hand, random variability may exist between individual observations which are nested in each subject. These represent the random effects, as shown in Table 5. In the <u><strong>initial model</strong></u>, Subject_ID was treated as the sole random effect. In the <u><strong>final model</strong></u>, both random effects were incorporated (Subject_ID, Observation_Number).</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>variables_table2 <span class="ot">&lt;-</span> variables_table <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Type =</span> <span class="fu">c</span>(</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Random"</span>,</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Random"</span>,</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span>,</span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Fixed"</span></span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>variables_table2 <span class="sc">%&gt;%</span></span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a>  gt <span class="sc">%&gt;%</span></span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tab_header</span>(</span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Table 5. Variable Categorization"</span></span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tab_footnote</span>(</span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">footnote =</span> <span class="st">"A break down of random and fixed effects based on the purpose of the study. Variable categorization is a crucial step in the LMM process."</span></span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<div id="cxxsuitzvv" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#cxxsuitzvv table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#cxxsuitzvv thead, #cxxsuitzvv tbody, #cxxsuitzvv tfoot, #cxxsuitzvv tr, #cxxsuitzvv td, #cxxsuitzvv th {
  border-style: none;
}

#cxxsuitzvv p {
  margin: 0;
  padding: 0;
}

#cxxsuitzvv .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#cxxsuitzvv .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#cxxsuitzvv .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#cxxsuitzvv .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#cxxsuitzvv .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#cxxsuitzvv .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cxxsuitzvv .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#cxxsuitzvv .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#cxxsuitzvv .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#cxxsuitzvv .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#cxxsuitzvv .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#cxxsuitzvv .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#cxxsuitzvv .gt_spanner_row {
  border-bottom-style: hidden;
}

#cxxsuitzvv .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#cxxsuitzvv .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#cxxsuitzvv .gt_from_md > :first-child {
  margin-top: 0;
}

#cxxsuitzvv .gt_from_md > :last-child {
  margin-bottom: 0;
}

#cxxsuitzvv .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#cxxsuitzvv .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#cxxsuitzvv .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#cxxsuitzvv .gt_row_group_first td {
  border-top-width: 2px;
}

#cxxsuitzvv .gt_row_group_first th {
  border-top-width: 2px;
}

#cxxsuitzvv .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#cxxsuitzvv .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#cxxsuitzvv .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#cxxsuitzvv .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cxxsuitzvv .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#cxxsuitzvv .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#cxxsuitzvv .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#cxxsuitzvv .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#cxxsuitzvv .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cxxsuitzvv .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#cxxsuitzvv .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#cxxsuitzvv .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#cxxsuitzvv .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#cxxsuitzvv .gt_left {
  text-align: left;
}

#cxxsuitzvv .gt_center {
  text-align: center;
}

#cxxsuitzvv .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#cxxsuitzvv .gt_font_normal {
  font-weight: normal;
}

#cxxsuitzvv .gt_font_bold {
  font-weight: bold;
}

#cxxsuitzvv .gt_font_italic {
  font-style: italic;
}

#cxxsuitzvv .gt_super {
  font-size: 65%;
}

#cxxsuitzvv .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#cxxsuitzvv .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#cxxsuitzvv .gt_indent_1 {
  text-indent: 5px;
}

#cxxsuitzvv .gt_indent_2 {
  text-indent: 10px;
}

#cxxsuitzvv .gt_indent_3 {
  text-indent: 15px;
}

#cxxsuitzvv .gt_indent_4 {
  text-indent: 20px;
}

#cxxsuitzvv .gt_indent_5 {
  text-indent: 25px;
}
</style>

<table class="gt_table table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="header gt_heading">
<th colspan="2" class="gt_heading gt_title gt_font_normal gt_bottom_border">Table 5. Variable Categorization</th>
</tr>
<tr class="odd gt_col_headings">
<th id="Variable" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Variable</th>
<th id="Type" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Type</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">Group</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">Subject_ID</td>
<td class="gt_row gt_left" headers="Type">Random</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">Observation_number</td>
<td class="gt_row gt_left" headers="Type">Random</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">Hydroxyurea</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">Asthma</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">ICS</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">LABA</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">Gender</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">Age_months</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">Height_cm</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">Weight_Kg</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">BMI</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">R5Hz_PP</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">R20Hz_PP</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Variable">X5Hz_PP</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Variable">Fres_PP</td>
<td class="gt_row gt_left" headers="Type">Fixed</td>
</tr>
</tbody><tfoot class="gt_footnotes">
<tr class="even">
<td colspan="2" class="gt_footnote">A break down of random and fixed effects based on the purpose of the study. Variable categorization is a crucial step in the LMM process.</td>
</tr>
</tfoot>

</table>

</div>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co">#lme()</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit models using a tidy and clear approach</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>model_lme <span class="ot">&lt;-</span> <span class="fu">lme</span>(</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">fixed =</span> <span class="fu">cbind</span>(R5Hz_PP, R20Hz_PP, X5Hz_PP, Fres_PP) <span class="sc">~</span> BMI <span class="sc">+</span> Asthma <span class="sc">+</span> ICS <span class="sc">+</span> LABA <span class="sc">+</span> Gender <span class="sc">+</span> Age_months <span class="sc">+</span> Height_cm <span class="sc">+</span> Weight_Kg,</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">random =</span> <span class="fu">list</span>(<span class="at">Subject_ID =</span> <span class="fu">pdIdent</span>(<span class="sc">~</span><span class="dv">1</span>)),</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> x_clean,</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"REML"</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="co">#lmer() </span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>model_lmer <span class="ot">&lt;-</span> <span class="fu">lmer</span>(</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> R5Hz_PP <span class="sc">+</span> R20Hz_PP <span class="sc">+</span> X5Hz_PP <span class="sc">+</span> Fres_PP <span class="sc">~</span> BMI <span class="sc">+</span> Asthma <span class="sc">+</span> ICS <span class="sc">+</span> LABA <span class="sc">+</span> Gender <span class="sc">+</span> Age_months <span class="sc">+</span> Height_cm <span class="sc">+</span> Weight_Kg <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> Subject_ID),</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> x_clean</span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="initial-model" class="level3">
<h3 class="anchored" data-anchor-id="initial-model"><strong>3.3.1 Initial Model</strong></h3>
<p><img src="images/initial_model.png" class="img-fluid quarto-figure quarto-figure-center" alt="Equation 1. The initial linear mixed model.">{.lightbox}</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare models based on AIC</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>aic_lme <span class="ot">&lt;-</span> <span class="fu">AIC</span>(model_lme)</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>aic_lmer <span class="ot">&lt;-</span> <span class="fu">AIC</span>(model_lmer)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"AIC for lme model: %f</span><span class="sc">\n</span><span class="st">"</span>, aic_lme))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>AIC for lme model: 1898.947925</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"AIC for lmer model: %f</span><span class="sc">\n</span><span class="st">"</span>, aic_lmer))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>AIC for lmer model: 2517.366883</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Correctly assign final_model based on AIC comparison</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (aic_lme <span class="sc">&lt;</span> aic_lmer) {</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  final_model <span class="ot">&lt;-</span> model_lme</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  model_type <span class="ot">&lt;-</span> <span class="st">"lme"</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  final_model <span class="ot">&lt;-</span> model_lmer</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  model_type <span class="ot">&lt;-</span> <span class="st">"lmer"</span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Final model selected: %s</span><span class="sc">\n</span><span class="st">"</span>, model_type))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Final model selected: lme</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Since final_model is now correctly assigned, we can call summary on it</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(final_model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed-effects model fit by REML
  Data: x_clean 
       AIC      BIC   logLik
  1898.948 1935.007 -938.474

Random effects:
 Formula: ~1 | Subject_ID
        (Intercept) Residual
StdDev:    20.45439 19.76738

Fixed effects:  cbind(R5Hz_PP, R20Hz_PP, X5Hz_PP, Fres_PP) ~ BMI + Asthma + ICS +      LABA + Gender + Age_months + Height_cm + Weight_Kg 
                Value Std.Error  DF   t-value p-value
(Intercept) 219.89260  86.92601 111  2.529653  0.0128
BMI          -5.89676   2.67942 111 -2.200761  0.0298
AsthmaYes    16.70033   7.68459  85  2.173222  0.0325
ICSYes       -4.93414   4.87163 111 -1.012830  0.3133
LABAYes      -4.35652   6.08228 111 -0.716265  0.4753
Gendermale   -5.68901   5.71803  85 -0.994924  0.3226
Age_months    0.49796   0.13627 111  3.654088  0.0004
Height_cm    -1.04120   0.60384 111 -1.724307  0.0874
Weight_Kg     1.65587   1.01041 111  1.638804  0.1041
 Correlation: 
           (Intr) BMI    AsthmY ICSYes LABAYs Gndrml Ag_mnt Hght_c
BMI        -0.927                                                 
AsthmaYes  -0.135  0.054                                          
ICSYes      0.115 -0.074 -0.378                                   
LABAYes    -0.011  0.026 -0.055 -0.298                            
Gendermale -0.146  0.146 -0.009 -0.070  0.037                     
Age_months  0.329 -0.101  0.049 -0.002  0.029  0.146              
Height_cm  -0.965  0.830  0.074 -0.110  0.002  0.050 -0.535       
Weight_Kg   0.938 -0.986 -0.082  0.087 -0.030 -0.122  0.087 -0.852

Standardized Within-Group Residuals:
       Min         Q1        Med         Q3        Max 
-2.8899152 -0.4840087 -0.1223361  0.4272441  2.5859876 

Number of Observations: 205
Number of Groups: 88 </code></pre>
</div>
</div>
<p><strong>Akaike Information Criterion (AIC)</strong></p>
<p>The AIC for both models was calculated. The AIC is a measure of the relative quality of statistical models for a given set of data. Lower AIC values indicate a model that better fits the data without unnecessary complexity.</p>
<p>Here, the AIC for lme was 1898.95 while lmer was 2517.37.</p>
<p>The model with the lower AIC was selected as the <strong>final model (lme)</strong> despite performance improvements offered by the lme4 package. <u>All additional models were lme.</u></p>
<p><strong>Residuals</strong></p>
<p>Residual plots (Residuals vs.&nbsp;Fitted Values) were created for the lme model to assess the goodness of fit in Figure 11. A horizontal line at y=0 was added as a reference. These plots help in identifying non-linearity, unequal variances, and outliers.</p>
<p>Based on the <strong>residual plot</strong>, the model has an ideal random pattern of scattered values with a few possible outliers.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Residuals</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>residuals_final <span class="ot">&lt;-</span> <span class="fu">resid</span>(final_model)</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate fitted values and residuals from the final model</span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>fitted_values <span class="ot">&lt;-</span> <span class="fu">fitted</span>(final_model)</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>residual_values <span class="ot">&lt;-</span> <span class="fu">residuals</span>(final_model)</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a data frame explicitly for plotting</span></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Fitted =</span> fitted_values, <span class="at">Residuals =</span> residual_values)</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting using ggplot2 for a more flexible and powerful approach</span></span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Residuals vs Fitted Values</span></span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="at">x =</span> Fitted, <span class="at">y =</span> Residuals)) <span class="sc">+</span></span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Fitted Values"</span>, <span class="at">y =</span> <span class="st">"Residuals"</span>, <span class="at">title =</span> <span class="st">"Figure 11. Residuals vs. Fitted Values"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/InitialResiduals-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Histogram of Residuals and QQ Plots</strong></p>
<p>A histogram and a Q-Q (Quantile-Quantile) plot of the residuals were used to check the normality assumption of the residuals (Figure 12). Finally, a QQ plot with a QQ line was produced for a graphical normality check (Figure 13).</p>
<p>Based on the <strong>histogram</strong>, the model <u>visually</u> had an ideal bell-shaped curve that resembles the normal distribution. Based on the <strong>QQ plot</strong>, the model <u>graphically</u> may have had some residuals that were not normally distributed toward the ends.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Histogram of Residuals</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="at">x =</span> Residuals)) <span class="sc">+</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Figure 12. Histogram of Residuals"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/InitialQQ-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Q-Q Plot</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(residuals_final, <span class="at">main =</span> <span class="st">"Figure 13. Q-Q Plot of Residuals"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/InitialQQ-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code> 35  40 
 95 113 </code></pre>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shapiro-Wilk Normality Test</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>shapiro_test_results <span class="ot">&lt;-</span> <span class="fu">shapiro.test</span>(residuals_final)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(shapiro_test_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
    Shapiro-Wilk normality test

data:  residuals_final
W = 0.95883, p-value = 1.163e-05</code></pre>
</div>
</div>
<p>The Shapiro-Wilk test was conducted on the residuals to formally test for normality.</p>
<p><span class="math inline">\(H_o\)</span>: the residuals are normally distributed.</p>
<p><span class="math inline">\(H_a\)</span>: the residuals are not normally distributed.</p>
<p><span class="math inline">\(\alpha\)</span> = 0.05</p>
<p>In this case, P = 0.00001163. P &lt; 0.05, so the null hypothesis was rejected, suggesting that the <strong>residuals were not normally distributed.</strong> This model does not satisfy the assumptions of LMMs.</p>
</section>
<section id="imputed-model" class="level3">
<h3 class="anchored" data-anchor-id="imputed-model"><strong>3.3.2 Imputed Model</strong></h3>
<p>Outliers (as mentioned above) were present in most variables, and the residuals of the initial model were not normally distributed. To improve model performance, outliers were imputed using the the threshold values. The model was then regenerated and assessed using the same metrics as above (Figures 14-17).</p>
<p>Figure 14. Box plots of numerical variables.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Copy the original dataset</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>x_clean_imputed <span class="ot">&lt;-</span> x_clean</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a function for Winsorization</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>winsorize <span class="ot">&lt;-</span> <span class="cf">function</span>(x, <span class="at">lower_percentile =</span> <span class="fl">0.10</span>, <span class="at">upper_percentile =</span> <span class="fl">0.90</span>) {</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>  lower_threshold <span class="ot">&lt;-</span> <span class="fu">quantile</span>(x, lower_percentile)</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>  upper_threshold <span class="ot">&lt;-</span> <span class="fu">quantile</span>(x, upper_percentile)</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>  x[x <span class="sc">&lt;</span> lower_threshold] <span class="ot">&lt;-</span> lower_threshold</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>  x[x <span class="sc">&gt;</span> upper_threshold] <span class="ot">&lt;-</span> upper_threshold</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(x)</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply imputation across numeric variables in the copied dataset</span></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>numeric_vars <span class="ot">&lt;-</span> <span class="fu">names</span>(x_clean_imputed <span class="sc">%&gt;%</span> <span class="fu">select_if</span>(is.numeric))</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (col <span class="cf">in</span> numeric_vars) {</span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>  x_clean_imputed[[col]] <span class="ot">&lt;-</span> <span class="fu">winsorize</span>(x_clean_imputed[[col]])</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualization with ggplot2</span></span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot boxplots for each numeric variable after imputation</span></span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (col <span class="cf">in</span> numeric_vars) {</span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> x_clean_imputed, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="sc">!!</span><span class="fu">sym</span>(col))) <span class="sc">+</span></span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">fill =</span> <span class="st">"skyblue"</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Boxplot of"</span>, col), <span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> col)</span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/AddressingAssumptionsAndRandomEffects-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/AddressingAssumptionsAndRandomEffects-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/AddressingAssumptionsAndRandomEffects-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/AddressingAssumptionsAndRandomEffects-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/AddressingAssumptionsAndRandomEffects-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/AddressingAssumptionsAndRandomEffects-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/AddressingAssumptionsAndRandomEffects-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/AddressingAssumptionsAndRandomEffects-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding a general title for the entire set of boxplots</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="co">#mtext("Figure 14. Box plots of numerical variables.", side=3, line=1, outer=TRUE, cex=1.5)</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Modeling with Imputed Data</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Refit the model using the lme function with the cleaned data</span></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>model_lme_imputed <span class="ot">&lt;-</span> <span class="fu">lme</span>(<span class="at">fixed =</span> <span class="fu">cbind</span>(R5Hz_PP, R20Hz_PP, X5Hz_PP, Fres_PP) <span class="sc">~</span> BMI <span class="sc">+</span> Asthma <span class="sc">+</span> ICS <span class="sc">+</span> LABA <span class="sc">+</span> Gender <span class="sc">+</span> Age_months <span class="sc">+</span> Height_cm <span class="sc">+</span> Weight_Kg,</span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">random =</span> <span class="fu">list</span>(<span class="at">Subject_ID =</span> <span class="fu">pdIdent</span>(<span class="sc">~</span><span class="dv">1</span>)),</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">data =</span> x_clean_imputed,</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">"REML"</span>)</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>aic_lme_imputed <span class="ot">&lt;-</span> <span class="fu">AIC</span>(model_lme_imputed)</span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"AIC for lme model: %f</span><span class="sc">\n</span><span class="st">"</span>, aic_lme_imputed))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>AIC for lme model: 1790.908121</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract residuals</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>residuals_imputed <span class="ot">&lt;-</span> <span class="fu">resid</span>(model_lme_imputed)</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Residuals vs Fitted Values Plot</span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">data.frame</span>(<span class="at">Fitted =</span> <span class="fu">fitted</span>(model_lme_imputed), <span class="at">Residuals =</span> residuals_imputed), <span class="fu">aes</span>(<span class="at">x =</span> Fitted, <span class="at">y =</span> Residuals)) <span class="sc">+</span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Fitted Values"</span>, <span class="at">y =</span> <span class="st">"Residuals"</span>, <span class="at">title =</span> <span class="st">"Figure 15. Residuals vs. Fitted Values"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/AddressingAssumptionsAndRandomEffects-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Histogram of Residuals</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">data.frame</span>(<span class="at">Residuals =</span> residuals_imputed), <span class="fu">aes</span>(<span class="at">x =</span> Residuals)) <span class="sc">+</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Figure 16. Histogram of Residuals"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/AddressingAssumptionsAndRandomEffects-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(residuals_imputed, <span class="at">main =</span> <span class="st">"Figure 17. Q-Q Plot of Residuals"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/AddressingAssumptionsAndRandomEffects-11.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>17 29 
39 72 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Q-Q Plot and Shapiro-Wilk Test</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>shapiro_test_results <span class="ot">&lt;-</span> <span class="fu">shapiro.test</span>(residuals_imputed)</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(shapiro_test_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
    Shapiro-Wilk normality test

data:  residuals_imputed
W = 0.98664, p-value = 0.05066</code></pre>
</div>
</div>
<p>The AIC was calculated as 1790.91, <strong>an improvement on the initial model.</strong></p>
<p>The Shapiro-Wilk test was conducted on the residuals to formally test for normality.</p>
<p><span class="math inline">\(H_o\)</span>: the residuals are normally distributed.</p>
<p><span class="math inline">\(H_a\)</span>: the residuals are not normally distributed.</p>
<p><span class="math inline">\(\alpha\)</span> = 0.05</p>
<p>In this case, P = 0.05066. P &gt;0.05, so we failed to reject the null hypothesis, suggesting that the <strong>residuals were normally distributed</strong> after threshold imputation. This model now satisfies the assumptions of LMMs.</p>
</section>
<section id="final-model" class="level3">
<h3 class="anchored" data-anchor-id="final-model"><strong>3.3.3 Final Model</strong></h3>
<p>This was a longitudinal study involving multiple observations for each subject over time, and subjects are grouped into two categories (children with sickle cell disease and African-American children with asthma). Thus, in this final model, we modeled <strong><code>Group</code></strong> as a fixed effect since we were interested in the effect of the group itself on the outcome. <strong><code>Subject_ID</code></strong> should be a random effect to account for the repeated measures within subjects, and <strong><code>Observation_number</code></strong> was included as a random slope within <strong><code>Subject_ID</code></strong> (i.e., nested within Subject_ID). The same visualizations and tests were completed to assess the LMM assumptions (Figures 18-20). The residuals show a random pattern (Figure 18), the histogram is approximately normal (Figure 19), and the qq plot follows a straight line (Figure 20), indicating normality.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>model_lme_imputed_final <span class="ot">&lt;-</span> <span class="fu">lme</span>(<span class="at">fixed =</span> <span class="fu">cbind</span>(R5Hz_PP, R20Hz_PP, X5Hz_PP, Fres_PP) <span class="sc">~</span> BMI <span class="sc">+</span> Asthma <span class="sc">+</span> ICS <span class="sc">+</span> LABA <span class="sc">+</span> Gender <span class="sc">+</span> Age_months <span class="sc">+</span> Height_cm <span class="sc">+</span> Weight_Kg <span class="sc">+</span> Group,</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">data =</span> x_clean_imputed,</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">random =</span> <span class="fu">list</span>(<span class="at">Subject_ID =</span> <span class="fu">pdIdent</span>(<span class="sc">~</span><span class="dv">1</span> <span class="sc">+</span> Observation_number)),</span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">"REML"</span>)</span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(model_lme_imputed_final)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>List of 18
 $ modelStruct :List of 1
  ..$ reStruct:List of 1
  .. ..$ Subject_ID:'pdIdent' with matrix num [1:6, 1:6] 1.99 0 0 0 0 ...
  .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. ..$ : chr [1:6] "(Intercept)" "Observation_number2" "Observation_number3" "Observation_number4" ...
  .. .. .. ..$ : chr [1:6] "(Intercept)" "Observation_number2" "Observation_number3" "Observation_number4" ...
  .. ..- attr(*, "settings")= int [1:4] 1 1 0 2
  .. ..- attr(*, "class")= chr "reStruct"
  .. ..- attr(*, "plen")= Named int 1
  .. .. ..- attr(*, "names")= chr "Subject_ID"
  ..- attr(*, "settings")= num [1:4] 1 0 1 2
  ..- attr(*, "class")= chr [1:3] "lmeStructInt" "lmeStruct" "modelStruct"
  ..- attr(*, "pmap")= logi [1, 1] TRUE
  .. ..- attr(*, "dimnames")=List of 2
  .. .. ..$ : NULL
  .. .. ..$ : chr "reStruct"
  ..- attr(*, "fixedSigma")= logi FALSE
 $ dims        :List of 5
  ..$ N    : int 205
  ..$ Q    : int 1
  ..$ qvec : Named num [1:3] 6 0 0
  .. ..- attr(*, "names")= chr [1:3] "Subject_ID" "" ""
  ..$ ngrps: Named int [1:3] 88 1 1
  .. ..- attr(*, "names")= chr [1:3] "Subject_ID" "X" "y"
  ..$ ncol : Named num [1:3] 6 10 1
  .. ..- attr(*, "names")= chr [1:3] "Subject_ID" "" ""
 $ contrasts   :List of 2
  ..$ Observation_number: num [1:6, 1:5] 0 1 0 0 0 0 0 0 1 0 ...
  .. ..- attr(*, "dimnames")=List of 2
  .. .. ..$ : chr [1:6] "1" "2" "3" "4" ...
  .. .. ..$ : chr [1:5] "2" "3" "4" "5" ...
  ..$ Group             : num [1:2, 1] 0 1
  .. ..- attr(*, "dimnames")=List of 2
  .. .. ..$ : chr [1:2] "C-Asthma" "C-SCD"
  .. .. ..$ : chr "C-SCD"
 $ coefficients:List of 2
  ..$ fixed : Named num [1:10] 154.99 -3.77 15.54 -3.26 -3.16 ...
  .. ..- attr(*, "names")= chr [1:10] "(Intercept)" "BMI" "AsthmaYes" "ICSYes" ...
  ..$ random:List of 1
  .. ..$ Subject_ID: num [1:88, 1:6] -3.56 8.76 -1.48 13.86 8.56 ...
  .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. ..$ : chr [1:88] "1" "2" "3" "4" ...
  .. .. .. ..$ : chr [1:6] "(Intercept)" "Observation_number2" "Observation_number3" "Observation_number4" ...
 $ varFix      : num [1:10, 1:10] 4621.9 -125.4 -56.5 29.6 -33.7 ...
  ..- attr(*, "dimnames")=List of 2
  .. ..$ : chr [1:10] "(Intercept)" "BMI" "AsthmaYes" "ICSYes" ...
  .. ..$ : chr [1:10] "(Intercept)" "BMI" "AsthmaYes" "ICSYes" ...
 $ sigma       : num 11
 $ apVar       : num [1:2, 1:2] 0.0107 -0.0118 -0.0118 0.0277
  ..- attr(*, "dimnames")=List of 2
  .. ..$ : chr [1:2] "reStruct.Subject_ID" "lSigma"
  .. ..$ : chr [1:2] "reStruct.Subject_ID" "lSigma"
  ..- attr(*, "Pars")= Named num [1:2] 2.74 2.4
  .. ..- attr(*, "names")= chr [1:2] "reStruct.Subject_ID" "lSigma"
  ..- attr(*, "natural")= logi TRUE
 $ logLik      : num -889
 $ numIter     : NULL
 $ groups      :'data.frame':   205 obs. of  1 variable:
  ..$ Subject_ID: Factor w/ 88 levels "1","2","3","4",..: 1 1 2 3 3 4 5 6 6 7 ...
 $ call        : language lme.formula(fixed = cbind(R5Hz_PP, R20Hz_PP, X5Hz_PP, Fres_PP) ~ BMI +      Asthma + ICS + LABA + Gender + Age_mo| __truncated__ ...
 $ terms       :Classes 'terms', 'formula'  language cbind(R5Hz_PP, R20Hz_PP, X5Hz_PP, Fres_PP) ~ BMI + Asthma + ICS + LABA +      Gender + Age_months + Height_cm + W| __truncated__
  .. ..- attr(*, "variables")= language list(cbind(R5Hz_PP, R20Hz_PP, X5Hz_PP, Fres_PP), BMI, Asthma, ICS, LABA,      Gender, Age_months, Height_cm, Weight_Kg, Group)
  .. ..- attr(*, "factors")= int [1:10, 1:9] 0 1 0 0 0 0 0 0 0 0 ...
  .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. ..$ : chr [1:10] "cbind(R5Hz_PP, R20Hz_PP, X5Hz_PP, Fres_PP)" "BMI" "Asthma" "ICS" ...
  .. .. .. ..$ : chr [1:9] "BMI" "Asthma" "ICS" "LABA" ...
  .. ..- attr(*, "term.labels")= chr [1:9] "BMI" "Asthma" "ICS" "LABA" ...
  .. ..- attr(*, "order")= int [1:9] 1 1 1 1 1 1 1 1 1
  .. ..- attr(*, "intercept")= int 1
  .. ..- attr(*, "response")= int 1
  .. ..- attr(*, ".Environment")=&lt;environment: R_GlobalEnv&gt; 
  .. ..- attr(*, "predvars")= language list(cbind(R5Hz_PP, R20Hz_PP, X5Hz_PP, Fres_PP), BMI, Asthma, ICS, LABA,      Gender, Age_months, Height_cm, Weight_Kg, Group)
  .. ..- attr(*, "dataClasses")= Named chr [1:10] "nmatrix.4" "numeric" "character" "character" ...
  .. .. ..- attr(*, "names")= chr [1:10] "cbind(R5Hz_PP, R20Hz_PP, X5Hz_PP, Fres_PP)" "BMI" "Asthma" "ICS" ...
 $ method      : chr "REML"
 $ fitted      : num [1:205, 1:2] 112 110 111 113 113 ...
  ..- attr(*, "dimnames")=List of 2
  .. ..$ : chr [1:205] "2" "3" "5" "6" ...
  .. ..$ : chr [1:2] "fixed" "Subject_ID"
 $ residuals   : num [1:205, 1:2] -9.09 -3.37 13.18 -4.1 4.14 ...
  ..- attr(*, "dimnames")=List of 2
  .. ..$ : chr [1:205] "2" "3" "5" "6" ...
  .. ..$ : chr [1:2] "fixed" "Subject_ID"
  ..- attr(*, "std")= num [1:205] 11 11 11 11 11 ...
 $ fixDF       :List of 2
  ..$ X    : Named num [1:10] 111 111 84 111 111 84 111 111 111 84
  .. ..- attr(*, "names")= chr [1:10] "(Intercept)" "BMI" "AsthmaYes" "ICSYes" ...
  ..$ terms: Named num [1:10] 111 111 84 111 111 84 111 111 111 84
  .. ..- attr(*, "names")= chr [1:10] "(Intercept)" "BMI" "Asthma" "ICS" ...
  ..- attr(*, "assign")=List of 10
  .. ..$ (Intercept): int 1
  .. ..$ BMI        : int 2
  .. ..$ Asthma     : int 3
  .. ..$ ICS        : int 4
  .. ..$ LABA       : int 5
  .. ..$ Gender     : int 6
  .. ..$ Age_months : int 7
  .. ..$ Height_cm  : int 8
  .. ..$ Weight_Kg  : int 9
  .. ..$ Group      : int 10
  ..- attr(*, "varFixFact")= num [1:10, 1:10] -1.95 -10.45 -3.1 -1.14 -0.55 ...
 $ na.action   : NULL
 $ data        :'data.frame':   205 obs. of  16 variables:
  ..$ Group             : Factor w/ 2 levels "C-Asthma","C-SCD": 2 2 2 2 2 2 2 2 2 2 ...
  ..$ Subject_ID        : Factor w/ 90 levels "1","2","3","4",..: 1 1 2 3 3 4 5 6 6 7 ...
  ..$ Observation_number: Factor w/ 6 levels "1","2","3","4",..: 2 3 1 1 2 2 1 1 2 1 ...
  ..$ Hydroxyurea       : chr [1:205] "Yes" "Yes" "No" "Yes" ...
  ..$ Asthma            : chr [1:205] "Yes" "Yes" "No" "Yes" ...
  ..$ ICS               : chr [1:205] "Yes" "Yes" "No" "No" ...
  ..$ LABA              : chr [1:205] "No" "Yes" "No" "No" ...
  ..$ Gender            : chr [1:205] "male" "male" "female" "male" ...
  ..$ Age_months        : num [1:205] 193 198 198 178 186 ...
  ..$ Height_cm         : num [1:205] 163 164 154 158 164 ...
  ..$ Weight_Kg         : num [1:205] 62.3 63.1 65.7 51.9 56.7 ...
  ..$ BMI               : num [1:205] 23.5 23.6 24.6 20.8 20.9 ...
  ..$ R5Hz_PP           : num [1:205] 103 107 124 109 117 146 136 96 123 117 ...
  ..$ R20Hz_PP          : num [1:205] 98 98 117 86 105 117 97 109 115 117 ...
  ..$ X5Hz_PP           : num [1:205] 111 174 98 115 107 195 140 111 164 92 ...
  ..$ Fres_PP           : num [1:205] 169 159 135 148 159 175 176 104 109 152 ...
  ..- attr(*, "na.action")= 'omit' Named int [1:14] 1 4 8 13 18 21 44 46 50 52 ...
  .. ..- attr(*, "names")= chr [1:14] "1" "4" "8" "13" ...
 - attr(*, "class")= chr "lme"</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>aic_lme_imputed_final <span class="ot">&lt;-</span> <span class="fu">AIC</span>(model_lme_imputed_final)</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"AIC for lme model: %f</span><span class="sc">\n</span><span class="st">"</span>, aic_lme_imputed_final))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>AIC for lme model: 1801.597804</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract residuals</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>residuals_imputed <span class="ot">&lt;-</span> <span class="fu">resid</span>(model_lme_imputed_final)</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Residuals vs Fitted Values Plot</span></span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">data.frame</span>(<span class="at">Fitted =</span> <span class="fu">fitted</span>(model_lme_imputed_final), <span class="at">Residuals =</span> residuals_imputed), <span class="fu">aes</span>(<span class="at">x =</span> Fitted, <span class="at">y =</span> Residuals)) <span class="sc">+</span></span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Fitted Values"</span>, <span class="at">y =</span> <span class="st">"Residuals"</span>, <span class="at">title =</span> <span class="st">"Figure 18. Residuals vs. Fitted Values"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Histogram of Residuals</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">data.frame</span>(<span class="at">Residuals =</span> residuals_imputed), <span class="fu">aes</span>(<span class="at">x =</span> Residuals)) <span class="sc">+</span></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">1</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Figure 19. Histogram of Residuals"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-21-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Q-Q Plot of Residuals</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(residuals_imputed, <span class="at">main =</span> <span class="st">"Figure 20. Q-Q Plot of Residuals"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-21-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>17 29 
39 72 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shapiro-Wilk Test for Normality of Residuals</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>shapiro_test_results <span class="ot">&lt;-</span> <span class="fu">shapiro.test</span>(residuals_imputed)</span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(shapiro_test_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
    Shapiro-Wilk normality test

data:  residuals_imputed
W = 0.98676, p-value = 0.0529</code></pre>
</div>
</div>
<p><img src="images/final_model.png" class="img-fluid quarto-figure quarto-figure-center" alt="Equation 2. The final linear mixed model.">{.lightbox}</p>
<p>The AIC was calculated as 1801.60, <strong>an improvement on the initial model but not on the less complex imputed model, as shown in Figure 21.</strong> The AIC penalizes model complexity to avoid overfitting, suggesting that the added effects of Group and Observation_number may not be sufficiently increasing model accuracy compared to complexity. However, these effects may still be relevant given the research goal of the project despite the slight increase in AIC, <strong>and thus will be left in the final model.</strong></p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Model names</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>model_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"1. LME Model"</span>, <span class="st">"2. LME Imputed Model"</span>, <span class="st">"3. LME Imputed Final Model"</span>)</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Combining into a dataframe</span></span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>aic_review <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">Model =</span> model_names,</span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">AIC =</span> <span class="fu">c</span>(aic_lme, aic_lme_imputed, aic_lme_imputed_final)</span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a>aic_review<span class="sc">$</span>Model <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(aic_review<span class="sc">$</span>Model)</span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a>aic_review<span class="sc">$</span>AIC <span class="ot">&lt;-</span> <span class="fu">round</span>(aic_review<span class="sc">$</span>AIC, <span class="dv">2</span>)</span>
<span id="cb67-12"><a href="#cb67-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-13"><a href="#cb67-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the structure</span></span>
<span id="cb67-14"><a href="#cb67-14" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(aic_review)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>'data.frame':   3 obs. of  2 variables:
 $ Model: Factor w/ 3 levels "1. LME Model",..: 1 2 3
 $ AIC  : num  1899 1791 1802</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(aic_review, <span class="fu">aes</span>(<span class="at">x =</span> Model, <span class="at">y =</span> AIC, <span class="at">fill =</span> Model)) <span class="sc">+</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Figure 21. AIC Values for Different Models"</span>,</span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Model"</span>,</span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"AIC Value"</span>) <span class="sc">+</span></span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> AIC), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.3</span>, <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The Shapiro-Wilk test was conducted on the residuals to formally test for normality.</p>
<p><span class="math inline">\(H_o\)</span>: the residuals are normally distributed.</p>
<p><span class="math inline">\(H_a\)</span>: the residuals are not normally distributed.</p>
<p><span class="math inline">\(\alpha\)</span> = 0.05</p>
<p>In this case, P = 0.0529. P &gt;0.05, so we failed to reject the null hypothesis, suggesting that the <strong>residuals were normally distributed</strong> after threshold imputation. This final model also satisfies the assumptions of LMMs.</p>
</section>
<section id="predictions" class="level3">
<h3 class="anchored" data-anchor-id="predictions"><strong>3.3.4 Predictions</strong></h3>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sctter plot of predicted and actuals on y axis and most important category on x axis </span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="co"># and split into groups </span></span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">43</span>)</span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>lme_resids <span class="ot">=</span> <span class="fu">residuals</span>(model_lme)</span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>lme_imputed_resids <span class="ot">=</span> <span class="fu">residuals</span>(model_lme_imputed)</span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a>lme_imputed_final_resids <span class="ot">=</span> <span class="fu">residuals</span>(model_lme_imputed_final)</span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a>lme_mse <span class="ot">=</span> <span class="fu">mean</span>(lme_resids<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a>lme_mae <span class="ot">=</span> <span class="fu">mean</span>(<span class="fu">abs</span>(lme_resids))</span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-17"><a href="#cb70-17" aria-hidden="true" tabindex="-1"></a>lme_imputed_mse <span class="ot">=</span> <span class="fu">mean</span>(lme_imputed_resids<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb70-18"><a href="#cb70-18" aria-hidden="true" tabindex="-1"></a>lme_imputed_mae <span class="ot">=</span> <span class="fu">mean</span>(<span class="fu">abs</span>(lme_imputed_resids))</span>
<span id="cb70-19"><a href="#cb70-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-20"><a href="#cb70-20" aria-hidden="true" tabindex="-1"></a>lme_imputed_final_mse <span class="ot">=</span> <span class="fu">mean</span>(lme_imputed_final_resids<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb70-21"><a href="#cb70-21" aria-hidden="true" tabindex="-1"></a>lme_imputed_final_mae <span class="ot">=</span> <span class="fu">mean</span>(<span class="fu">abs</span>(lme_imputed_final_resids))</span>
<span id="cb70-22"><a href="#cb70-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-23"><a href="#cb70-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-24"><a href="#cb70-24" aria-hidden="true" tabindex="-1"></a>mse_review <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb70-25"><a href="#cb70-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">Model =</span> model_names,</span>
<span id="cb70-26"><a href="#cb70-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">MSE =</span> <span class="fu">c</span>(lme_mse, lme_imputed_mse, lme_imputed_final_mse)</span>
<span id="cb70-27"><a href="#cb70-27" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb70-28"><a href="#cb70-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-29"><a href="#cb70-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-30"><a href="#cb70-30" aria-hidden="true" tabindex="-1"></a>mse_review<span class="sc">$</span>MSE <span class="ot">&lt;-</span> <span class="fu">round</span>(mse_review<span class="sc">$</span>MSE, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb70-31"><a href="#cb70-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-32"><a href="#cb70-32" aria-hidden="true" tabindex="-1"></a>mae_review <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb70-33"><a href="#cb70-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">Model =</span> model_names,</span>
<span id="cb70-34"><a href="#cb70-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">MAE =</span> <span class="fu">c</span>(lme_mae, lme_imputed_mae, lme_imputed_final_mae)</span>
<span id="cb70-35"><a href="#cb70-35" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb70-36"><a href="#cb70-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-37"><a href="#cb70-37" aria-hidden="true" tabindex="-1"></a>mae_review<span class="sc">$</span>MAE <span class="ot">&lt;-</span> <span class="fu">round</span>(mae_review<span class="sc">$</span>MAE, <span class="at">digits =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>MSE and MAE</strong></p>
<p><strong>Mean Squared Error (MSE)</strong> and <strong>Mean Absolute Error (MAE)</strong> are metrics used to asses the performance of a model. MSE is the mean of the individual residuals squared and MAE is the mean of the individual absolute value of the residuals. As shown in figures 22 and 23, the imputed final model out performs the other two models by a significant margin. It is important to note that MSE is impacted more by larger errors or outliers because it squares the residuals</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(mse_review, <span class="fu">aes</span>(<span class="at">x =</span> Model, <span class="at">y =</span> MSE, <span class="at">fill =</span> Model)) <span class="sc">+</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Figure 22. MSE Values for Different Models"</span>,</span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Model"</span>,</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"MSE Value"</span>) <span class="sc">+</span></span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> MSE), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.3</span>, <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(mae_review, <span class="fu">aes</span>(<span class="at">x =</span> Model, <span class="at">y =</span> MAE, <span class="at">fill =</span> Model)) <span class="sc">+</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Figure 23. MAE Values for Different Models"</span>,</span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Model"</span>,</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"MAE Value"</span>) <span class="sc">+</span></span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> MAE), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.3</span>, <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Sample Predictions vs Actual</strong></p>
<p>The bar graph below compares the <u>actual</u> <code>R5Hz_PP</code> to <u>predicted</u> <code>R5Hz_PP</code> (as a measure of airway resistance and reactance) for <strong>10 random subjects</strong>. The difference in the bars for each subject is the <strong>residual error.</strong> The small residual error present for each subject suggests that the model is accurate at predicting <code>R5Hz_PP</code> as a measure of airway resistance and reactance.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>lme_imputed_final_predictions <span class="ot">=</span> <span class="fu">predict</span>(model_lme_imputed_final)</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>lme_imputed_fina_preds_actuals <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(lme_imputed_final_predictions, x_clean_imputed<span class="sc">$</span>R5Hz_PP))</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(lme_imputed_fina_preds_actuals) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Predicted_R5Hz_PP"</span>, <span class="st">"Actual_R5Hz_PP"</span> )</span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>sample_indices <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">nrow</span>(x_clean_imputed), <span class="dv">10</span>)</span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a>sample_pred_actuals <span class="ot">=</span> lme_imputed_fina_preds_actuals[sample_indices, ]</span>
<span id="cb73-10"><a href="#cb73-10" aria-hidden="true" tabindex="-1"></a>sample_pred_actuals<span class="sc">$</span>row <span class="ot">&lt;-</span><span class="dv">1</span><span class="sc">:</span><span class="dv">10</span></span>
<span id="cb73-11"><a href="#cb73-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-12"><a href="#cb73-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-13"><a href="#cb73-13" aria-hidden="true" tabindex="-1"></a>sample_pred_actuals_melt <span class="ot">&lt;-</span> <span class="fu">melt</span>(sample_pred_actuals, <span class="at">id.vars =</span> <span class="st">"row"</span>)</span>
<span id="cb73-14"><a href="#cb73-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-15"><a href="#cb73-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-16"><a href="#cb73-16" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sample_pred_actuals_melt, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(row), <span class="at">y =</span> value, <span class="at">fill =</span> variable)) <span class="sc">+</span></span>
<span id="cb73-17"><a href="#cb73-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb73-18"><a href="#cb73-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Observation"</span>, <span class="at">y =</span> <span class="st">"R5Hz_PP"</span>, <span class="at">fill =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb73-19"><a href="#cb73-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb73-20"><a href="#cb73-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"top"</span>) <span class="sc">+</span></span>
<span id="cb73-21"><a href="#cb73-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Figure 24. Sample Comparison of Predicted and Actual Values"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="conclusion" class="level1">
<h1><strong>4 Conclusion</strong></h1>
<p>Linear mixed models are versatile tools for modeling complex relations with multiple effects (fixed and random), as well as missing and non-independent data. For the given capstone dataset, the final linear mixed model can reliably predict measures of airway resistance and reactance given demographic and co-morbidity data. This model can be reliably used for both children with Sickle Cell Disease and those with asthma to provide insights into their respiratory function.</p>
</section>
<section id="references" class="level1">
<h1><strong>5 References</strong></h1>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-aarts_2015" class="csl-entry" role="listitem">
Aarts, E., Dolan, C. V., Verhage, M., and Sluis, S. van der (2015), <span>“Multilevel analysis quantifies variation in the experimental effect while optimizing power and preventing false positives,”</span> <em>BMC Neuroscience</em>, 16, 94. <a href="https://doi.org/10.1186/s12868-015-0228-5">https://doi.org/10.1186/s12868-015-0228-5</a>.
</div>
<div id="ref-barr_random_2013" class="csl-entry" role="listitem">
Barr, D. J. (2013), <span>“Random effects structure for testing interactions in linear mixed-effects models,”</span> <em>Frontiers in Psychology</em>, 4, 328. <a href="https://doi.org/10.3389/fpsyg.2013.00328">https://doi.org/10.3389/fpsyg.2013.00328</a>.
</div>
<div id="ref-bates_fitting_2015" class="csl-entry" role="listitem">
Bates, D., Mächler, M., Bolker, B., and Walker, S. (2015), <span>“Fitting <span>Linear</span> <span>Mixed</span>-<span>Effects</span> <span>Models</span> <span>Using</span> lme4,”</span> <em>Journal of Statistical Software</em>, 67, 1–48. <a href="https://doi.org/10.18637/jss.v067.i01">https://doi.org/10.18637/jss.v067.i01</a>.
</div>
<div id="ref-bolker_generalized_2009" class="csl-entry" role="listitem">
Bolker, B. M., Brooks, M. E., Clark, C. J., Geange, S. W., Poulsen, J. R., Stevens, M. H. H., and White, J.-S. S. (2009), <span>“Generalized linear mixed models: A practical guide for ecology and evolution,”</span> <em>Trends in Ecology &amp; Evolution</em>, 24, 127–135. <a href="https://doi.org/10.1016/j.tree.2008.10.008">https://doi.org/10.1016/j.tree.2008.10.008</a>.
</div>
<div id="ref-brown_introduction_2021" class="csl-entry" role="listitem">
Brown, V. A. (2021), <span>“An <span>Introduction</span> to <span>Linear</span> <span>Mixed</span>-<span>Effects</span> <span>Modeling</span> in <span>R</span>,”</span> <em>Advances in Methods and Practices in Psychological Science</em>, 4, 2515245920960351. <a href="https://doi.org/10.1177/2515245920960351">https://doi.org/10.1177/2515245920960351</a>.
</div>
<div id="ref-casals_methodological_2014" class="csl-entry" role="listitem">
Casals, M., Girabent-Farrés, M., and Carrasco, J. L. (2014), <span>“Methodological quality and reporting of generalized linear mixed models in clinical medicine (2000-2012): A systematic review,”</span> <em>PloS One</em>, 9, e112653. <a href="https://doi.org/10.1371/journal.pone.0112653">https://doi.org/10.1371/journal.pone.0112653</a>.
</div>
<div id="ref-galecki_linear_2014" class="csl-entry" role="listitem">
Galecki, A. T., Kathleen B. Welch (2014), <em>Linear <span>Mixed</span> <span>Models</span>: <span>A</span> <span>Practical</span> <span>Guide</span> <span>Using</span> <span>Statistical</span> <span>Software</span>, <span>Second</span> <span>Edition</span></em>, New York: Chapman; Hall/CRC. <a href="https://doi.org/10.1201/b17198">https://doi.org/10.1201/b17198</a>.
</div>
<div id="ref-grueber_2011" class="csl-entry" role="listitem">
GRUEBER, C. E., NAKAGAWA, S., LAWS, R. J., and JAMIESON, I. G. (2011), <span>“Multimodel inference in ecology and evolution: Challenges and solutions,”</span> <em>Journal of Evolutionary Biology</em>, 24, 699–711. https://doi.org/<a href="https://doi.org/10.1111/j.1420-9101.2010.02210.x">https://doi.org/10.1111/j.1420-9101.2010.02210.x</a>.
</div>
<div id="ref-harrison_brief_2018" class="csl-entry" role="listitem">
Harrison, X. A., Donaldson, L., Correa-Cano, M. E., Evans, J., Fisher, D. N., Goodwin, C. E. D., Robinson, B. S., Hodgson, D. J., and Inger, R. (2018), <span>“A brief introduction to mixed effects modelling and multi-model inference in ecology,”</span> <em>PeerJ</em>, 6, e4794. <a href="https://doi.org/10.7717/peerj.4794">https://doi.org/10.7717/peerj.4794</a>.
</div>
<div id="ref-jolly_pymer4_2018" class="csl-entry" role="listitem">
Jolly, E. (2018), <span>“Pymer4: <span>Connecting</span> <span>R</span> and <span>Python</span> for <span>Linear</span> <span>Mixed</span> <span>Modeling</span>,”</span> <em>Journal of Open Source Software</em>, 3, 862. <a href="https://doi.org/10.21105/joss.00862">https://doi.org/10.21105/joss.00862</a>.
</div>
<div id="ref-lee_estimation_nodate" class="csl-entry" role="listitem">
Lee, Y.-C., and Shang, J. (n.d.), <span>“Estimation and selection in linear mixed models with missing data under compound symmetric structure,”</span> <em>Journal of Applied Statistics</em>, 49, 4003–4027. <a href="https://doi.org/10.1080/02664763.2021.1969342">https://doi.org/10.1080/02664763.2021.1969342</a>.
</div>
<div id="ref-lo_transform_2015" class="csl-entry" role="listitem">
Lo, S., and Andrews, S. (2015), <span>“<a href="https://www.frontiersin.org/journals/psychology/articles/10.3389/fpsyg.2015.01171">To transform or not to transform: Using generalized linear mixed models to analyse reaction time data</a>,”</span> <em>Frontiers in Psychology</em>, 6.
</div>
<div id="ref-magezi_linear_2015" class="csl-entry" role="listitem">
Magezi, D. A. (2015), <span>“Linear mixed-effects models for within-participant psychology experiments: An introductory tutorial and free, graphical user interface (<span>LMMgui</span>),”</span> <em>Frontiers in Psychology</em>, 6, 2. <a href="https://doi.org/10.3389/fpsyg.2015.00002">https://doi.org/10.3389/fpsyg.2015.00002</a>.
</div>
<div id="ref-fokkema_generalized_2021" class="csl-entry" role="listitem">
Marjolein Fokkema, J. E.-C., and Wolpert, M. (2021), <span>“Generalized linear mixed-model (GLMM) trees: A flexible decision-tree method for multilevel and longitudinal data,”</span> <em>Psychotherapy Research</em>, Routledge, 31, 329–341. <a href="https://doi.org/10.1080/10503307.2020.1785037">https://doi.org/10.1080/10503307.2020.1785037</a>.
</div>
<div id="ref-monsalves_level_2020" class="csl-entry" role="listitem">
Monsalves, M. J., Bangdiwala, A. S., Thabane, A., and Bangdiwala, S. I. (2020), <span>“<span>LEVEL</span> (<span>Logical</span> <span>Explanations</span> &amp; <span>Visualizations</span> of <span>Estimates</span> in <span>Linear</span> mixed models): Recommendations for reporting multilevel data and analyses,”</span> <em>BMC Medical Research Methodology</em>, 20, 3. <a href="https://doi.org/10.1186/s12874-019-0876-8">https://doi.org/10.1186/s12874-019-0876-8</a>.
</div>
<div id="ref-peng_model_2012" class="csl-entry" role="listitem">
Peng, H., and Lu, Y. (2012), <span>“Model selection in linear mixed effect models,”</span> <em>Journal of Multivariate Analysis</em>, 109, 109–129. <a href="https://doi.org/10.1016/j.jmva.2012.02.005">https://doi.org/10.1016/j.jmva.2012.02.005</a>.
</div>
<div id="ref-piepho_analysing_1999" class="csl-entry" role="listitem">
Piepho (1999), <span>“Analysing disease incidence data from designed experiments by generalized linear mixed models,”</span> <em>Plant Pathology</em>, 48, 668–674. https://doi.org/<a href="https://doi.org/10.1046/j.1365-3059.1999.00383.x">https://doi.org/10.1046/j.1365-3059.1999.00383.x</a>.
</div>
<div id="ref-pusponegoro_linear_2017" class="csl-entry" role="listitem">
Pusponegoro, N. H., Rachmawati, R. N., Notodiputro, K. A., and Sartono, B. (2017), <span>“Linear <span>Mixed</span> <span>Model</span> for <span>Analyzing</span> <span>Longitudinal</span> <span>Data</span>: <span>A</span> <span>Simulation</span> <span>Study</span> of <span>Children</span> <span>Growth</span> <span>Differences</span>,”</span> <em>Procedia Computer Science</em>, Discovery and innovation of computer science technology in artificial intelligence era: <span>The</span> 2nd <span>International</span> <span>Conference</span> on <span>Computer</span> <span>Science</span> and <span>Computational</span> <span>Intelligence</span> (<span>ICCSCI</span> 2017), 116, 284–291. <a href="https://doi.org/10.1016/j.procs.2017.10.071">https://doi.org/10.1016/j.procs.2017.10.071</a>.
</div>
<div id="ref-schielzeth_robustness_2020" class="csl-entry" role="listitem">
Schielzeth, H., Dingemanse, N. J., Nakagawa, S., Westneat, D. F., Allegue, H., Teplitsky, C., Réale, D., Dochtermann, N. A., Garamszegi, L. Z., and Araya-Ajoy, Y. G. (2020), <span>“Robustness of linear mixed-effects models to violations of distributional assumptions,”</span> <em>Methods in Ecology and Evolution</em>, 11, 1141–1152. https://doi.org/<a href="https://doi.org/10.1111/2041-210X.13434">https://doi.org/10.1111/2041-210X.13434</a>.
</div>
<div id="ref-steibel_powerful_2009" class="csl-entry" role="listitem">
Steibel, J. P., Poletto, R., Coussens, P. M., and Rosa, G. J. M. (2009), <span>“A powerful and flexible linear mixed model framework for the analysis of relative quantification <span>RT</span>-<span>PCR</span> data,”</span> <em>Genomics</em>, 94, 146–152. <a href="https://doi.org/10.1016/j.ygeno.2009.04.008">https://doi.org/10.1016/j.ygeno.2009.04.008</a>.
</div>
<div id="ref-touraine_when_2023" class="csl-entry" role="listitem">
Touraine, C., Cuer, B., Conroy, T., Juzyna, B., Gourgou, S., and Mollevi, C. (2023), <span>“When a joint model should be preferred over a linear mixed model for analysis of longitudinal health-related quality of life data in cancer clinical trials,”</span> <em>BMC medical research methodology</em>, 23, 36. <a href="https://doi.org/10.1186/s12874-023-01846-3">https://doi.org/10.1186/s12874-023-01846-3</a>.
</div>
<div id="ref-tu_using_2015" class="csl-entry" role="listitem">
Tu, Y.-K. (2015), <span>“Using generalized linear mixed models to evaluate inconsistency within a network meta-analysis,”</span> <em>Value in Health</em>, 18, 1120–1125. https://doi.org/<a href="https://doi.org/10.1016/j.jval.2015.10.002">https://doi.org/10.1016/j.jval.2015.10.002</a>.
</div>
<div id="ref-verbeeck_linear_2023" class="csl-entry" role="listitem">
Verbeeck, J., Faes, C., Neyens, T., Hens, N., Verbeke, G., Deboosere, P., and Molenberghs, G. (2023), <span>“A linear mixed model to estimate <span>COVID</span>-19-induced excess mortality,”</span> <em>Biometrics</em>, 79, 417–425. <a href="https://doi.org/10.1111/biom.13578">https://doi.org/10.1111/biom.13578</a>.
</div>
<div id="ref-wang_statistical_2022" class="csl-entry" role="listitem">
Wang, X., Andrinopoulou, E.-R., Veen, K. M., Bogers, A. J. J. C., and Takkenberg, J. J. M. (2022), <span>“Statistical primer: An introduction to the application of linear mixed-effects models in cardiothoracic surgery outcomes research-a case study using homograft pulmonary valve replacement data,”</span> <em>European Journal of Cardio-Thoracic Surgery: Official Journal of the European Association for Cardio-Thoracic Surgery</em>, 62, ezac429. <a href="https://doi.org/10.1093/ejcts/ezac429">https://doi.org/10.1093/ejcts/ezac429</a>.
</div>
<div id="ref-zuur_2016" class="csl-entry" role="listitem">
Zuur, A. F., and Ieno, E. N. (2016), <span>“A protocol for conducting and presenting results of regression-type analyses,”</span> <em>Methods in Ecology and Evolution</em>, 7, 636–645. https://doi.org/<a href="https://doi.org/10.1111/2041-210X.12577">https://doi.org/10.1111/2041-210X.12577</a>.
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>